--- net/minecraft/client/renderer/debug/BeeDebugRenderer.java
+++ net/minecraft/client/renderer/debug/BeeDebugRenderer.java
@@ -178,11 +178,11 @@
 
    private static String getBeeUuidsAsString(Collection<UUID> var0) {
       if (var0.isEmpty()) {
          return "-";
       } else {
-         return var0.size() > 3 ? "" + var0.size() + " bees" : ((Set)var0.stream().map(DebugEntityNameGenerator::getEntityName).collect(Collectors.toSet())).toString();
+         return var0.size() > 3 ? var0.size() + " bees" : ((Set)var0.stream().map(DebugEntityNameGenerator::getEntityName).collect(Collectors.toSet())).toString();
       }
    }
 
    private static void highlightHive(BlockPos var0) {
       float var1 = 0.05F;
@@ -190,11 +190,11 @@
    }
 
    private void renderGhostHive(BlockPos var1, List<String> var2) {
       float var3 = 0.05F;
       renderTransparentFilledBox(var1, 0.05F, 0.2F, 0.2F, 1.0F, 0.3F);
-      renderTextOverPos("" + var2, var1, 0, -256);
+      renderTextOverPos(var2.makeConcatWithConstants<invokedynamic>(var2), var1, 0, -256);
       renderTextOverPos("Ghost Hive", var1, 1, -65536);
    }
 
    private static void renderTransparentFilledBox(BlockPos var0, float var1, float var2, float var3, float var4, float var5) {
       RenderSystem.enableBlend();
@@ -215,11 +215,12 @@
          renderTextOverHive("In: 1 bee", var1, var3++, -256);
       } else {
          renderTextOverHive("In: " + var1.occupantCount + " bees", var1, var3++, -256);
       }
 
-      renderTextOverHive("Honey: " + var1.honeyLevel, var1, var3++, -23296);
+      int var4 = var1.honeyLevel;
+      renderTextOverHive("Honey: " + var4, var1, var3++, -23296);
       renderTextOverHive(var1.hiveType + (var1.sedated ? " (sedated)" : ""), var1, var3++, -1);
    }
 
    private void renderPath(BeeDebugRenderer.BeeInfo var1) {
       if (var1.path != null) {
@@ -297,11 +298,12 @@
    }
 
    private String getPosDescription(BeeDebugRenderer.BeeInfo var1, BlockPos var2) {
       float var3 = Mth.sqrt(var2.distSqr(var1.pos.x(), var1.pos.y(), var1.pos.z(), true));
       double var4 = (double)Math.round(var3 * 10.0F) / 10.0D;
-      return var2.toShortString() + " (dist " + var4 + ")";
+      String var10000 = var2.toShortString();
+      return var10000 + " (dist " + var4 + ")";
    }
 
    private boolean isBeeSelected(BeeDebugRenderer.BeeInfo var1) {
       return Objects.equals(this.lastLookedAtUuid, var1.uuid);
    }
@@ -339,10 +341,28 @@
       DebugRenderer.getTargetedEntity(this.minecraft.getCameraEntity(), 8).ifPresent((var1) -> {
          this.lastLookedAtUuid = var1.getUUID();
       });
    }
 
+   public static class HiveInfo {
+      public final BlockPos pos;
+      public final String hiveType;
+      public final int occupantCount;
+      public final int honeyLevel;
+      public final boolean sedated;
+      public final long lastSeen;
+
+      public HiveInfo(BlockPos var1, String var2, int var3, int var4, boolean var5, long var6) {
+         this.pos = var1;
+         this.hiveType = var2;
+         this.occupantCount = var3;
+         this.honeyLevel = var4;
+         this.sedated = var5;
+         this.lastSeen = var6;
+      }
+   }
+
    public static class BeeInfo {
       public final UUID uuid;
       public final int id;
       public final Position pos;
       @Nullable
@@ -383,24 +403,6 @@
 
       public boolean hasFlower() {
          return this.flowerPos != null;
       }
    }
-
-   public static class HiveInfo {
-      public final BlockPos pos;
-      public final String hiveType;
-      public final int occupantCount;
-      public final int honeyLevel;
-      public final boolean sedated;
-      public final long lastSeen;
-
-      public HiveInfo(BlockPos var1, String var2, int var3, int var4, boolean var5, long var6) {
-         this.pos = var1;
-         this.hiveType = var2;
-         this.occupantCount = var3;
-         this.honeyLevel = var4;
-         this.sedated = var5;
-         this.lastSeen = var6;
-      }
-   }
 }
