--- net/minecraft/client/renderer/debug/GameEventListenerRenderer.java
+++ net/minecraft/client/renderer/debug/GameEventListenerRenderer.java
@@ -142,27 +142,11 @@
 
    public void trackListener(PositionSource var1, int var2) {
       this.trackedListeners.add(new GameEventListenerRenderer.TrackedListener(var1, var2));
    }
 
-   static class TrackedGameEvent {
-      public final long timeStamp;
-      public final GameEvent gameEvent;
-      public final Vec3 position;
-
-      public TrackedGameEvent(long var1, GameEvent var3, Vec3 var4) {
-         this.timeStamp = var1;
-         this.gameEvent = var3;
-         this.position = var4;
-      }
-
-      public boolean isExpired() {
-         return Util.getMillis() - this.timeStamp > 3000L;
-      }
-   }
-
-   static class TrackedListener implements GameEventListener {
+   private static class TrackedListener implements GameEventListener {
       public final PositionSource listenerSource;
       public final int listenerRange;
 
       public TrackedListener(PositionSource var1, int var2) {
          this.listenerSource = var1;
@@ -188,6 +172,22 @@
 
       public boolean handleGameEvent(Level var1, GameEvent var2, @Nullable Entity var3, BlockPos var4) {
          return false;
       }
    }
+
+   private static class TrackedGameEvent {
+      public final long timeStamp;
+      public final GameEvent gameEvent;
+      public final Vec3 position;
+
+      public TrackedGameEvent(long var1, GameEvent var3, Vec3 var4) {
+         this.timeStamp = var1;
+         this.gameEvent = var3;
+         this.position = var4;
+      }
+
+      public boolean isExpired() {
+         return Util.getMillis() - this.timeStamp > 3000L;
+      }
+   }
 }
