--- net/minecraft/client/gui/screens/recipebook/GhostRecipe.java
+++ net/minecraft/client/gui/screens/recipebook/GhostRecipe.java
@@ -15,11 +15,11 @@
 import net.minecraft.world.item.crafting.Recipe;
 
 public class GhostRecipe {
    private Recipe<?> recipe;
    private final List<GhostRecipe.GhostIngredient> ingredients = Lists.newArrayList();
-   private float time;
+   float time;
 
    public GhostRecipe() {
    }
 
    public void clear() {
