--- net/minecraft/util/profiling/SingleTickProfiler.java
+++ net/minecraft/util/profiling/SingleTickProfiler.java
@@ -34,11 +34,14 @@
    public void endTick() {
       if (this.profiler != InactiveProfiler.INSTANCE) {
          ProfileResults var1 = this.profiler.getResults();
          this.profiler = InactiveProfiler.INSTANCE;
          if (var1.getNanoDuration() >= this.saveThreshold) {
-            File var2 = new File(this.location, "tick-results-" + (new SimpleDateFormat("yyyy-MM-dd_HH.mm.ss")).format(new Date()) + ".txt");
+            File var10002 = this.location;
+            SimpleDateFormat var10003 = new SimpleDateFormat("yyyy-MM-dd_HH.mm.ss");
+            Date var10004 = new Date();
+            File var2 = new File(var10002, "tick-results-" + var10003.format(var10004) + ".txt");
             var1.saveResults(var2.toPath());
             LOGGER.info((String)"Recorded long tick -- wrote info to: {}", (Object)var2.getAbsolutePath());
          }
 
       }
