--- net/minecraft/world/entity/npc/WanderingTrader.java
+++ net/minecraft/world/entity/npc/WanderingTrader.java
@@ -210,15 +210,15 @@
    public void setWanderTarget(@Nullable BlockPos var1) {
       this.wanderTarget = var1;
    }
 
    @Nullable
-   private BlockPos getWanderTarget() {
+   BlockPos getWanderTarget() {
       return this.wanderTarget;
    }
 
-   class WanderToPositionGoal extends Goal {
+   private class WanderToPositionGoal extends Goal {
       final WanderingTrader trader;
       final double stopDistance;
       final double speedModifier;
 
       WanderToPositionGoal(WanderingTrader var2, double var3, double var5) {
