--- com/mojang/blaze3d/vertex/PoseStack.java
+++ com/mojang/blaze3d/vertex/PoseStack.java
@@ -75,14 +75,14 @@
    public void mulPoseMatrix(Matrix4f var1) {
       ((PoseStack.Pose)this.poseStack.getLast()).pose.multiply(var1);
    }
 
    public static final class Pose {
-      private final Matrix4f pose;
-      private final Matrix3f normal;
+      final Matrix4f pose;
+      final Matrix3f normal;
 
-      private Pose(Matrix4f var1, Matrix3f var2) {
+      Pose(Matrix4f var1, Matrix3f var2) {
          this.pose = var1;
          this.normal = var2;
       }
 
       public Matrix4f pose() {
