--- com/mojang/blaze3d/vertex/VertexBuffer.java
+++ com/mojang/blaze3d/vertex/VertexBuffer.java
@@ -3,10 +3,11 @@
 import com.mojang.blaze3d.platform.Window;
 import com.mojang.blaze3d.systems.RenderSystem;
 import com.mojang.datafixers.util.Pair;
 import com.mojang.math.Matrix4f;
 import java.nio.ByteBuffer;
+import java.util.Objects;
 import java.util.concurrent.CompletableFuture;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.ShaderInstance;
 
 public class VertexBuffer implements AutoCloseable {
@@ -63,10 +64,11 @@
    public CompletableFuture<Void> uploadLater(BufferBuilder var1) {
       if (!RenderSystem.isOnRenderThread()) {
          return CompletableFuture.runAsync(() -> {
             this.upload_(var1);
          }, (var0) -> {
+            Objects.requireNonNull(var0);
             RenderSystem.recordRenderCall(var0::run);
          });
       } else {
          this.upload_(var1);
          return CompletableFuture.completedFuture((Object)null);
