--- net/minecraft/world/level/block/entity/AbstractFurnaceBlockEntity.java
+++ net/minecraft/world/level/block/entity/AbstractFurnaceBlockEntity.java
@@ -58,14 +58,14 @@
    public static final int DATA_COOKING_TOTAL_TIME = 3;
    public static final int NUM_DATA_VALUES = 4;
    public static final int BURN_TIME_STANDARD = 200;
    public static final int BURN_COOL_SPEED = 2;
    protected NonNullList<ItemStack> items;
-   private int litTime;
-   private int litDuration;
-   private int cookingProgress;
-   private int cookingTotalTime;
+   int litTime;
+   int litDuration;
+   int cookingProgress;
+   int cookingTotalTime;
    protected final ContainerData dataAccess;
    private final Object2IntOpenHashMap<ResourceLocation> recipesUsed;
    private final RecipeType<? extends AbstractCookingRecipe> recipeType;
 
    protected AbstractFurnaceBlockEntity(BlockEntityType<?> var1, BlockPos var2, BlockState var3, RecipeType<? extends AbstractCookingRecipe> var4) {
