--- net/minecraft/network/chat/NbtComponent.java
+++ net/minecraft/network/chat/NbtComponent.java
@@ -142,11 +142,12 @@
             return Objects.equals(this.id, var2.id) && Objects.equals(this.nbtPathPattern, var2.nbtPathPattern) && super.equals(var1);
          }
       }
 
       public String toString() {
-         return "StorageNbtComponent{id='" + this.id + '\'' + "path='" + this.nbtPathPattern + '\'' + ", siblings=" + this.siblings + ", style=" + this.getStyle() + '}';
+         ResourceLocation var10000 = this.id;
+         return "StorageNbtComponent{id='" + var10000 + "'path='" + this.nbtPathPattern + "', siblings=" + this.siblings + ", style=" + this.getStyle() + "}";
       }
    }
 
    public static class BlockNbtComponent extends NbtComponent {
       private final String posPattern;
@@ -208,11 +209,12 @@
             return Objects.equals(this.posPattern, var2.posPattern) && Objects.equals(this.nbtPathPattern, var2.nbtPathPattern) && super.equals(var1);
          }
       }
 
       public String toString() {
-         return "BlockPosArgument{pos='" + this.posPattern + '\'' + "path='" + this.nbtPathPattern + '\'' + ", siblings=" + this.siblings + ", style=" + this.getStyle() + '}';
+         String var10000 = this.posPattern;
+         return "BlockPosArgument{pos='" + var10000 + "'path='" + this.nbtPathPattern + "', siblings=" + this.siblings + ", style=" + this.getStyle() + "}";
       }
    }
 
    public static class EntityNbtComponent extends NbtComponent {
       private final String selectorPattern;
@@ -268,9 +270,10 @@
             return Objects.equals(this.selectorPattern, var2.selectorPattern) && Objects.equals(this.nbtPathPattern, var2.nbtPathPattern) && super.equals(var1);
          }
       }
 
       public String toString() {
-         return "EntityNbtComponent{selector='" + this.selectorPattern + '\'' + "path='" + this.nbtPathPattern + '\'' + ", siblings=" + this.siblings + ", style=" + this.getStyle() + '}';
+         String var10000 = this.selectorPattern;
+         return "EntityNbtComponent{selector='" + var10000 + "'path='" + this.nbtPathPattern + "', siblings=" + this.siblings + ", style=" + this.getStyle() + "}";
       }
    }
 }
