--- net/minecraft/network/chat/SelectorComponent.java
+++ net/minecraft/network/chat/SelectorComponent.java
@@ -73,8 +73,9 @@
          return this.pattern.equals(var2.pattern) && super.equals(var1);
       }
    }
 
    public String toString() {
-      return "SelectorComponent{pattern='" + this.pattern + '\'' + ", siblings=" + this.siblings + ", style=" + this.getStyle() + '}';
+      String var10000 = this.pattern;
+      return "SelectorComponent{pattern='" + var10000 + "', siblings=" + this.siblings + ", style=" + this.getStyle() + "}";
    }
 }
