--- net/minecraft/client/gui/components/BossHealthOverlay.java
+++ net/minecraft/client/gui/components/BossHealthOverlay.java
@@ -3,10 +3,11 @@
 import com.google.common.collect.Maps;
 import com.mojang.blaze3d.systems.RenderSystem;
 import com.mojang.blaze3d.vertex.PoseStack;
 import java.util.Iterator;
 import java.util.Map;
+import java.util.Objects;
 import java.util.UUID;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiComponent;
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.FormattedText;
@@ -18,11 +19,11 @@
    private static final ResourceLocation GUI_BARS_LOCATION = new ResourceLocation("textures/gui/bars.png");
    private static final int BAR_WIDTH = 182;
    private static final int BAR_HEIGHT = 5;
    private static final int OVERLAY_OFFSET = 80;
    private final Minecraft minecraft;
-   private final Map<UUID, LerpingBossEvent> events = Maps.newLinkedHashMap();
+   final Map<UUID, LerpingBossEvent> events = Maps.newLinkedHashMap();
 
    public BossHealthOverlay(Minecraft var1) {
       this.minecraft = var1;
    }
 
@@ -41,11 +42,11 @@
             Component var8 = var5.getName();
             int var9 = this.minecraft.font.width((FormattedText)var8);
             int var10 = var2 / 2 - var9 / 2;
             int var11 = var3 - 9;
             this.minecraft.font.drawShadow(var1, var8, (float)var10, (float)var11, 16777215);
-            this.minecraft.font.getClass();
+            Objects.requireNonNull(this.minecraft.font);
             var3 += 10 + 9;
             if (var3 >= this.minecraft.getWindow().getGuiScaledHeight() / 3) {
                break;
             }
          }
