--- net/minecraft/client/gui/components/ChatComponent.java
+++ net/minecraft/client/gui/components/ChatComponent.java
@@ -5,10 +5,11 @@
 import com.mojang.blaze3d.systems.RenderSystem;
 import com.mojang.blaze3d.vertex.PoseStack;
 import java.util.Deque;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Objects;
 import javax.annotation.Nullable;
 import net.minecraft.client.GuiMessage;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiComponent;
 import net.minecraft.client.gui.screens.ChatScreen;
@@ -101,11 +102,11 @@
                var1.popPose();
                RenderSystem.disableBlend();
             }
 
             if (var5) {
-               this.minecraft.font.getClass();
+               Objects.requireNonNull(this.minecraft.font);
                byte var27 = 9;
                var28 = var4 * var27;
                var19 = var16 * var27;
                int var29 = this.chatScrollbarPos * var19 / var4;
                int var21 = var19 * var19 / var28;
@@ -252,13 +253,13 @@
          var5 = (double)Mth.floor(var5 / this.getScale());
          var7 = (double)Mth.floor(var7 / (this.getScale() * (this.minecraft.options.chatLineSpacing + 1.0D)));
          if (var5 >= 0.0D && var7 >= 0.0D) {
             int var9 = Math.min(this.getLinesPerPage(), this.trimmedMessages.size());
             if (var5 <= (double)Mth.floor((double)this.getWidth() / this.getScale())) {
-               this.minecraft.font.getClass();
+               Objects.requireNonNull(this.minecraft.font);
                if (var7 < (double)(9 * var9 + var9)) {
-                  this.minecraft.font.getClass();
+                  Objects.requireNonNull(this.minecraft.font);
                   int var10 = (int)(var7 / 9.0D + (double)this.chatScrollbarPos);
                   if (var10 >= 0 && var10 < this.trimmedMessages.size()) {
                      GuiMessage var11 = (GuiMessage)this.trimmedMessages.get(var10);
                      return this.minecraft.font.getSplitter().componentStyleAtWidth((FormattedCharSequence)var11.getMessage(), (int)var5);
                   }
