--- net/minecraft/world/level/levelgen/VerticalAnchor.java
+++ net/minecraft/world/level/levelgen/VerticalAnchor.java
@@ -56,27 +56,27 @@
       CODEC = ExtraCodecs.xor(VerticalAnchor.Absolute.CODEC, ExtraCodecs.xor(VerticalAnchor.AboveBottom.CODEC, VerticalAnchor.BelowTop.CODEC)).xmap(VerticalAnchor::merge, VerticalAnchor::split);
       BOTTOM = aboveBottom(0);
       TOP = belowTop(0);
    }
 
-   static final class BelowTop extends VerticalAnchor {
-      public static final Codec<VerticalAnchor.BelowTop> CODEC;
+   static final class Absolute extends VerticalAnchor {
+      public static final Codec<VerticalAnchor.Absolute> CODEC;
 
-      protected BelowTop(int var1) {
+      protected Absolute(int var1) {
          super(var1);
       }
 
       public int resolveY(WorldGenerationContext var1) {
-         return var1.getGenDepth() - 1 + var1.getMinGenY() - this.value();
+         return this.value();
       }
 
       public String toString() {
-         return this.value() + " below top";
+         return this.value() + " absolute";
       }
 
       static {
-         CODEC = Codec.intRange(DimensionType.MIN_Y, DimensionType.MAX_Y).fieldOf("below_top").xmap(VerticalAnchor.BelowTop::new, VerticalAnchor::value).codec();
+         CODEC = Codec.intRange(DimensionType.MIN_Y, DimensionType.MAX_Y).fieldOf("absolute").xmap(VerticalAnchor.Absolute::new, VerticalAnchor::value).codec();
       }
    }
 
    static final class AboveBottom extends VerticalAnchor {
       public static final Codec<VerticalAnchor.AboveBottom> CODEC;
@@ -96,25 +96,25 @@
       static {
          CODEC = Codec.intRange(DimensionType.MIN_Y, DimensionType.MAX_Y).fieldOf("above_bottom").xmap(VerticalAnchor.AboveBottom::new, VerticalAnchor::value).codec();
       }
    }
 
-   static final class Absolute extends VerticalAnchor {
-      public static final Codec<VerticalAnchor.Absolute> CODEC;
+   private static final class BelowTop extends VerticalAnchor {
+      public static final Codec<VerticalAnchor.BelowTop> CODEC;
 
-      protected Absolute(int var1) {
+      protected BelowTop(int var1) {
          super(var1);
       }
 
       public int resolveY(WorldGenerationContext var1) {
-         return this.value();
+         return var1.getGenDepth() - 1 + var1.getMinGenY() - this.value();
       }
 
       public String toString() {
-         return this.value() + " absolute";
+         return this.value() + " below top";
       }
 
       static {
-         CODEC = Codec.intRange(DimensionType.MIN_Y, DimensionType.MAX_Y).fieldOf("absolute").xmap(VerticalAnchor.Absolute::new, VerticalAnchor::value).codec();
+         CODEC = Codec.intRange(DimensionType.MIN_Y, DimensionType.MAX_Y).fieldOf("below_top").xmap(VerticalAnchor.BelowTop::new, VerticalAnchor::value).codec();
       }
    }
 }
