--- net/minecraft/world/inventory/AnvilMenu.java
+++ net/minecraft/world/inventory/AnvilMenu.java
@@ -18,11 +18,11 @@
 import org.apache.logging.log4j.Logger;
 
 public class AnvilMenu extends ItemCombinerMenu {
    private static final Logger LOGGER = LogManager.getLogger();
    private static final boolean DEBUG_COST = false;
-   public static final int MAX_NAME_LENGTH = 35;
+   public static final int MAX_NAME_LENGTH = 50;
    private int repairItemCountCost;
    private String itemName;
    private final DataSlot cost;
    private static final int COST_FAIL = 0;
    private static final int COST_BASE = 1;
