--- net/minecraft/world/inventory/BeaconMenu.java
+++ net/minecraft/world/inventory/BeaconMenu.java
@@ -149,11 +149,11 @@
 
    public boolean hasPayment() {
       return !this.beacon.getItem(0).isEmpty();
    }
 
-   class PaymentSlot extends Slot {
+   private class PaymentSlot extends Slot {
       public PaymentSlot(Container var2, int var3, int var4, int var5) {
          super(var2, var3, var4, var5);
       }
 
       public boolean mayPlace(ItemStack var1) {
