--- net/minecraft/core/MappedRegistry.java
+++ net/minecraft/core/MappedRegistry.java
@@ -123,11 +123,11 @@
    public ResourceLocation getKey(T var1) {
       return (ResourceLocation)this.storage.inverse().get(var1);
    }
 
    public Optional<ResourceKey<T>> getResourceKey(T var1) {
-      return Optional.ofNullable(this.keyStorage.inverse().get(var1));
+      return Optional.ofNullable((ResourceKey)this.keyStorage.inverse().get(var1));
    }
 
    public int getId(@Nullable T var1) {
       return this.toId.getInt(var1);
    }
