--- net/minecraft/world/entity/EntitySelector.java
+++ net/minecraft/world/entity/EntitySelector.java
@@ -3,10 +3,11 @@
 import com.google.common.base.Predicates;
 import java.util.function.Predicate;
 import javax.annotation.Nullable;
 import net.minecraft.world.Container;
 import net.minecraft.world.Difficulty;
+import net.minecraft.world.entity.animal.axolotl.Axolotl;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.scores.Team;
 
 public final class EntitySelector {
@@ -22,11 +23,11 @@
    };
    public static final Predicate<Entity> NO_CREATIVE_OR_SPECTATOR = (var0) -> {
       return !(var0 instanceof Player) || !var0.isSpectator() && !((Player)var0).isCreative();
    };
    public static final Predicate<Entity> ATTACK_ALLOWED = (var0) -> {
-      return !(var0 instanceof Player) || !var0.isSpectator() && !((Player)var0).isCreative() && var0.level.getDifficulty() != Difficulty.PEACEFUL;
+      return (!(var0 instanceof Player) || !var0.isSpectator() && !((Player)var0).isCreative() && var0.level.getDifficulty() != Difficulty.PEACEFUL) && (!(var0 instanceof Axolotl) || !((Axolotl)var0).isPlayingDead());
    };
    public static final Predicate<Entity> NO_SPECTATORS = (var0) -> {
       return !var0.isSpectator();
    };
 
