--- net/minecraft/network/chat/ScoreComponent.java
+++ net/minecraft/network/chat/ScoreComponent.java
@@ -106,8 +106,9 @@
          return this.name.equals(var2.name) && this.objective.equals(var2.objective) && super.equals(var1);
       }
    }
 
    public String toString() {
-      return "ScoreComponent{name='" + this.name + '\'' + "objective='" + this.objective + '\'' + ", siblings=" + this.siblings + ", style=" + this.getStyle() + '}';
+      String var10000 = this.name;
+      return "ScoreComponent{name='" + var10000 + "'objective='" + this.objective + "', siblings=" + this.siblings + ", style=" + this.getStyle() + "}";
    }
 }
