--- net/minecraft/resources/ResourceLocation.java
+++ net/minecraft/resources/ResourceLocation.java
@@ -30,13 +30,13 @@
 
    protected ResourceLocation(String[] var1) {
       this.namespace = StringUtils.isEmpty(var1[0]) ? "minecraft" : var1[0];
       this.path = var1[1];
       if (!isValidNamespace(this.namespace)) {
-         throw new ResourceLocationException("Non [a-z0-9_.-] character in namespace of location: " + this.namespace + ':' + this.path);
+         throw new ResourceLocationException("Non [a-z0-9_.-] character in namespace of location: " + this.namespace + ":" + this.path);
       } else if (!isValidPath(this.path)) {
-         throw new ResourceLocationException("Non [a-z0-9/._-] character in path of location: " + this.namespace + ':' + this.path);
+         throw new ResourceLocationException("Non [a-z0-9/._-] character in path of location: " + this.namespace + ":" + this.path);
       }
    }
 
    public ResourceLocation(String var1) {
       this(decompose(var1, ':'));
@@ -87,11 +87,11 @@
    public String getNamespace() {
       return this.namespace;
    }
 
    public String toString() {
-      return this.namespace + ':' + this.path;
+      return this.namespace + ":" + this.path;
    }
 
    public boolean equals(Object var1) {
       if (this == var1) {
          return true;
