--- net/minecraft/world/level/levelgen/heightproviders/VeryBiasedToBottomHeight.java
+++ net/minecraft/world/level/levelgen/heightproviders/VeryBiasedToBottomHeight.java
@@ -51,8 +51,8 @@
    public HeightProviderType<?> getType() {
       return HeightProviderType.VERY_BIASED_TO_BOTTOM;
    }
 
    public String toString() {
-      return "biased[" + this.minInclusive + '-' + this.maxInclusive + " inner: " + this.inner + "]";
+      return "biased[" + this.minInclusive + "-" + this.maxInclusive + " inner: " + this.inner + "]";
    }
 }
