--- net/minecraft/advancements/AdvancementList.java
+++ net/minecraft/advancements/AdvancementList.java
@@ -4,10 +4,11 @@
 import com.google.common.collect.Sets;
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Set;
 import java.util.Map.Entry;
 import javax.annotation.Nullable;
 import net.minecraft.resources.ResourceLocation;
 import org.apache.logging.log4j.LogManager;
@@ -74,11 +75,11 @@
          while(var4.hasNext()) {
             var5 = (Entry)var4.next();
             ResourceLocation var6 = (ResourceLocation)var5.getKey();
             Advancement.Builder var7 = (Advancement.Builder)var5.getValue();
             Map var10001 = this.advancements;
-            var10001.getClass();
+            Objects.requireNonNull(var10001);
             if (var7.canBuild(var10001::get)) {
                Advancement var8 = var7.build(var6);
                this.advancements.put(var6, var8);
                var3 = true;
                var4.remove();
