--- com/mojang/blaze3d/platform/GlStateManager.java
+++ com/mojang/blaze3d/platform/GlStateManager.java
@@ -715,59 +715,17 @@
    public static int _getInteger(int var0) {
       RenderSystem.assertThread(RenderSystem::isOnRenderThreadOrInit);
       return GL11.glGetInteger(var0);
    }
 
-   @DontObfuscate
-   public static enum DestFactor {
-      CONSTANT_ALPHA(32771),
-      CONSTANT_COLOR(32769),
-      DST_ALPHA(772),
-      DST_COLOR(774),
-      ONE(1),
-      ONE_MINUS_CONSTANT_ALPHA(32772),
-      ONE_MINUS_CONSTANT_COLOR(32770),
-      ONE_MINUS_DST_ALPHA(773),
-      ONE_MINUS_DST_COLOR(775),
-      ONE_MINUS_SRC_ALPHA(771),
-      ONE_MINUS_SRC_COLOR(769),
-      SRC_ALPHA(770),
-      SRC_COLOR(768),
-      ZERO(0);
+   static class ScissorState {
+      public final GlStateManager.BooleanState mode = new GlStateManager.BooleanState(3089);
 
-      public final int value;
-
-      private DestFactor(int var3) {
-         this.value = var3;
+      ScissorState() {
       }
    }
 
-   @DontObfuscate
-   public static enum SourceFactor {
-      CONSTANT_ALPHA(32771),
-      CONSTANT_COLOR(32769),
-      DST_ALPHA(772),
-      DST_COLOR(774),
-      ONE(1),
-      ONE_MINUS_CONSTANT_ALPHA(32772),
-      ONE_MINUS_CONSTANT_COLOR(32770),
-      ONE_MINUS_DST_ALPHA(773),
-      ONE_MINUS_DST_COLOR(775),
-      ONE_MINUS_SRC_ALPHA(771),
-      ONE_MINUS_SRC_COLOR(769),
-      SRC_ALPHA(770),
-      SRC_ALPHA_SATURATE(776),
-      SRC_COLOR(768),
-      ZERO(0);
-
-      public final int value;
-
-      private SourceFactor(int var3) {
-         this.value = var3;
-      }
-   }
-
    static class BooleanState {
       private final int state;
       private boolean enabled;
 
       public BooleanState(int var1) {
@@ -794,124 +752,61 @@
          }
 
       }
    }
 
-   static class ColorMask {
-      public boolean red;
-      public boolean green;
-      public boolean blue;
-      public boolean alpha;
+   static class DepthState {
+      public final GlStateManager.BooleanState mode = new GlStateManager.BooleanState(2929);
+      public boolean mask = true;
+      public int func = 513;
 
-      private ColorMask() {
-         this.red = true;
-         this.green = true;
-         this.blue = true;
-         this.alpha = true;
+      DepthState() {
       }
    }
 
-   static class ScissorState {
-      public final GlStateManager.BooleanState mode;
+   static class BlendState {
+      public final GlStateManager.BooleanState mode = new GlStateManager.BooleanState(3042);
+      public int srcRgb = 1;
+      public int dstRgb = 0;
+      public int srcAlpha = 1;
+      public int dstAlpha = 0;
 
-      private ScissorState() {
-         this.mode = new GlStateManager.BooleanState(3089);
+      BlendState() {
       }
    }
 
-   static class StencilState {
-      public final GlStateManager.StencilFunc func;
-      public int mask;
-      public int fail;
-      public int zfail;
-      public int zpass;
+   static class CullState {
+      public final GlStateManager.BooleanState enable = new GlStateManager.BooleanState(2884);
+      public int mode = 1029;
 
-      private StencilState() {
-         this.func = new GlStateManager.StencilFunc();
-         this.mask = -1;
-         this.fail = 7680;
-         this.zfail = 7680;
-         this.zpass = 7680;
+      CullState() {
       }
    }
 
-   static class StencilFunc {
-      public int func;
-      public int ref;
-      public int mask;
-
-      private StencilFunc() {
-         this.func = 519;
-         this.mask = -1;
-      }
-   }
-
-   static class ColorLogicState {
-      public final GlStateManager.BooleanState enable;
-      public int op;
-
-      private ColorLogicState() {
-         this.enable = new GlStateManager.BooleanState(3058);
-         this.op = 5379;
-      }
-   }
-
    static class PolygonOffsetState {
-      public final GlStateManager.BooleanState fill;
-      public final GlStateManager.BooleanState line;
+      public final GlStateManager.BooleanState fill = new GlStateManager.BooleanState(32823);
+      public final GlStateManager.BooleanState line = new GlStateManager.BooleanState(10754);
       public float factor;
       public float units;
 
-      private PolygonOffsetState() {
-         this.fill = new GlStateManager.BooleanState(32823);
-         this.line = new GlStateManager.BooleanState(10754);
+      PolygonOffsetState() {
       }
    }
 
-   static class CullState {
-      public final GlStateManager.BooleanState enable;
-      public int mode;
+   static class ColorLogicState {
+      public final GlStateManager.BooleanState enable = new GlStateManager.BooleanState(3058);
+      public int op = 5379;
 
-      private CullState() {
-         this.enable = new GlStateManager.BooleanState(2884);
-         this.mode = 1029;
+      ColorLogicState() {
       }
    }
 
-   static class DepthState {
-      public final GlStateManager.BooleanState mode;
-      public boolean mask;
-      public int func;
-
-      private DepthState() {
-         this.mode = new GlStateManager.BooleanState(2929);
-         this.mask = true;
-         this.func = 513;
-      }
-   }
-
-   static class BlendState {
-      public final GlStateManager.BooleanState mode;
-      public int srcRgb;
-      public int dstRgb;
-      public int srcAlpha;
-      public int dstAlpha;
-
-      private BlendState() {
-         this.mode = new GlStateManager.BooleanState(3042);
-         this.srcRgb = 1;
-         this.dstRgb = 0;
-         this.srcAlpha = 1;
-         this.dstAlpha = 0;
-      }
-   }
-
    static class TextureState {
       public boolean enable;
       public int binding;
 
-      private TextureState() {
+      TextureState() {
       }
    }
 
    public static enum Viewport {
       INSTANCE;
@@ -939,10 +834,89 @@
       public static int height() {
          return INSTANCE.height;
       }
    }
 
+   static class ColorMask {
+      public boolean red = true;
+      public boolean green = true;
+      public boolean blue = true;
+      public boolean alpha = true;
+
+      ColorMask() {
+      }
+   }
+
+   static class StencilState {
+      public final GlStateManager.StencilFunc func = new GlStateManager.StencilFunc();
+      public int mask = -1;
+      public int fail = 7680;
+      public int zfail = 7680;
+      public int zpass = 7680;
+
+      StencilState() {
+      }
+   }
+
+   private static class StencilFunc {
+      public int func = 519;
+      public int ref;
+      public int mask = -1;
+
+      StencilFunc() {
+      }
+   }
+
+   @DontObfuscate
+   public static enum DestFactor {
+      CONSTANT_ALPHA(32771),
+      CONSTANT_COLOR(32769),
+      DST_ALPHA(772),
+      DST_COLOR(774),
+      ONE(1),
+      ONE_MINUS_CONSTANT_ALPHA(32772),
+      ONE_MINUS_CONSTANT_COLOR(32770),
+      ONE_MINUS_DST_ALPHA(773),
+      ONE_MINUS_DST_COLOR(775),
+      ONE_MINUS_SRC_ALPHA(771),
+      ONE_MINUS_SRC_COLOR(769),
+      SRC_ALPHA(770),
+      SRC_COLOR(768),
+      ZERO(0);
+
+      public final int value;
+
+      private DestFactor(int var3) {
+         this.value = var3;
+      }
+   }
+
+   @DontObfuscate
+   public static enum SourceFactor {
+      CONSTANT_ALPHA(32771),
+      CONSTANT_COLOR(32769),
+      DST_ALPHA(772),
+      DST_COLOR(774),
+      ONE(1),
+      ONE_MINUS_CONSTANT_ALPHA(32772),
+      ONE_MINUS_CONSTANT_COLOR(32770),
+      ONE_MINUS_DST_ALPHA(773),
+      ONE_MINUS_DST_COLOR(775),
+      ONE_MINUS_SRC_ALPHA(771),
+      ONE_MINUS_SRC_COLOR(769),
+      SRC_ALPHA(770),
+      SRC_ALPHA_SATURATE(776),
+      SRC_COLOR(768),
+      ZERO(0);
+
+      public final int value;
+
+      private SourceFactor(int var3) {
+         this.value = var3;
+      }
+   }
+
    public static enum LogicOp {
       AND(5377),
       AND_INVERTED(5380),
       AND_REVERSE(5378),
       CLEAR(5376),
