--- com/mojang/blaze3d/platform/TextureUtil.java
+++ com/mojang/blaze3d/platform/TextureUtil.java
@@ -136,35 +136,28 @@
          int var7 = var3 >> var5;
          int var8 = var4 >> var5;
 
          try {
             NativeImage var9 = new NativeImage(var7, var8, false);
-            Throwable var10 = null;
 
             try {
                var9.downloadTexture(var5, false);
                var9.writeToFile(var6);
                LOGGER.debug("Exported png to: {}", (new File(var6)).getAbsolutePath());
-            } catch (Throwable var20) {
-               var10 = var20;
-               throw var20;
-            } finally {
-               if (var9 != null) {
-                  if (var10 != null) {
-                     try {
-                        var9.close();
-                     } catch (Throwable var19) {
-                        var10.addSuppressed(var19);
-                     }
-                  } else {
-                     var9.close();
-                  }
+            } catch (Throwable var13) {
+               try {
+                  var9.close();
+               } catch (Throwable var12) {
+                  var13.addSuppressed(var12);
                }
 
+               throw var13;
             }
-         } catch (IOException var22) {
-            LOGGER.debug("Unable to write: ", var22);
+
+            var9.close();
+         } catch (IOException var14) {
+            LOGGER.debug("Unable to write: ", var14);
          }
       }
 
    }
 
