--- net/minecraft/world/level/dimension/LevelStem.java
+++ net/minecraft/world/level/dimension/LevelStem.java
@@ -65,11 +65,11 @@
 
       while(var2.hasNext()) {
          Entry var5 = (Entry)var2.next();
          ResourceKey var6 = (ResourceKey)var5.getKey();
          if (!BUILTIN_ORDER.contains(var6)) {
-            var1.register(var6, var5.getValue(), var0.lifecycle(var5.getValue()));
+            var1.register(var6, (LevelStem)var5.getValue(), var0.lifecycle((LevelStem)var5.getValue()));
          }
       }
 
       return var1;
    }
