--- net/minecraft/commands/synchronization/ArgumentTypes.java
+++ net/minecraft/commands/synchronization/ArgumentTypes.java
@@ -258,11 +258,11 @@
    static class Entry<T extends ArgumentType<?>> {
       public final Class<T> clazz;
       public final ArgumentSerializer<T> serializer;
       public final ResourceLocation name;
 
-      private Entry(Class<T> var1, ArgumentSerializer<T> var2, ResourceLocation var3) {
+      Entry(Class<T> var1, ArgumentSerializer<T> var2, ResourceLocation var3) {
          this.clazz = var1;
          this.serializer = var2;
          this.name = var3;
       }
    }
