--- net/minecraft/client/resources/sounds/BiomeAmbientSoundsHandler.java
+++ net/minecraft/client/resources/sounds/BiomeAmbientSoundsHandler.java
@@ -47,11 +47,11 @@
          this.previousBiome = var1;
          this.moodSettings = var1.getAmbientMood();
          this.additionsSettings = var1.getAmbientAdditions();
          this.loopSounds.values().forEach(BiomeAmbientSoundsHandler.LoopSoundInstance::fadeOut);
          var1.getAmbientLoop().ifPresent((var2) -> {
-            BiomeAmbientSoundsHandler.LoopSoundInstance var10000 = (BiomeAmbientSoundsHandler.LoopSoundInstance)this.loopSounds.compute(var1, (var2x, var3) -> {
+            this.loopSounds.compute(var1, (var2x, var3) -> {
                if (var3 == null) {
                   var3 = new BiomeAmbientSoundsHandler.LoopSoundInstance(var2);
                   this.soundManager.play(var3);
                }
 
