--- net/minecraft/world/level/levelgen/feature/structures/FeaturePoolElement.java
+++ net/minecraft/world/level/levelgen/feature/structures/FeaturePoolElement.java
@@ -73,8 +73,9 @@
    public StructurePoolElementType<?> getType() {
       return StructurePoolElementType.FEATURE;
    }
 
    public String toString() {
-      return "Feature[" + Registry.FEATURE.getKey(((ConfiguredFeature)this.feature.get()).feature()) + "]";
+      Registry var10000 = Registry.FEATURE;
+      return "Feature[" + var10000.getKey(((ConfiguredFeature)this.feature.get()).feature()) + "]";
    }
 }
