--- net/minecraft/world/level/entity/EntitySectionStorage.java
+++ net/minecraft/world/level/entity/EntitySectionStorage.java
@@ -79,11 +79,11 @@
    }
 
    public Stream<EntitySection<T>> getExistingSectionsInChunk(long var1) {
       LongStream var10000 = this.getExistingSectionPositionsInChunk(var1);
       Long2ObjectMap var10001 = this.sections;
-      var10001.getClass();
+      Objects.requireNonNull(var10001);
       return var10000.mapToObj(var10001::get).filter(Objects::nonNull);
    }
 
    private static long getChunkKeyFromSectionKey(long var0) {
       return ChunkPos.asLong(SectionPos.x(var0), SectionPos.z(var0));
