--- net/minecraft/client/tutorial/Tutorial.java
+++ net/minecraft/client/tutorial/Tutorial.java
@@ -98,13 +98,11 @@
       });
       var1.hide();
    }
 
    public void tick() {
-      this.timedToasts.removeIf((var0) -> {
-         return ((Tutorial.TimedToast)var0).updateProgress();
-      });
+      this.timedToasts.removeIf(Tutorial.TimedToast::updateProgress);
       if (this.instance != null) {
          if (this.minecraft.level != null) {
             this.instance.tick();
          } else {
             this.stop();
@@ -144,15 +142,15 @@
    public void onInventoryAction(ItemStack var1, ItemStack var2, ClickAction var3) {
       this.bundleTutorial.onInventoryAction(var1, var2, var3);
    }
 
    static final class TimedToast {
-      private final TutorialToast toast;
+      final TutorialToast toast;
       private final int durationTicks;
       private int progress;
 
-      private TimedToast(TutorialToast var1, int var2) {
+      TimedToast(TutorialToast var1, int var2) {
          this.toast = var1;
          this.durationTicks = var2;
       }
 
       private boolean updateProgress() {
