--- net/minecraft/data/advancements/AdvancementProvider.java
+++ net/minecraft/data/advancements/AdvancementProvider.java
@@ -52,11 +52,12 @@
       }
 
    }
 
    private static Path createPath(Path var0, Advancement var1) {
-      return var0.resolve("data/" + var1.getId().getNamespace() + "/advancements/" + var1.getId().getPath() + ".json");
+      String var10001 = var1.getId().getNamespace();
+      return var0.resolve("data/" + var10001 + "/advancements/" + var1.getId().getPath() + ".json");
    }
 
    public String getName() {
       return "Advancements";
    }
