--- net/minecraft/world/level/block/DoubleBlockCombiner.java
+++ net/minecraft/world/level/block/DoubleBlockCombiner.java
@@ -78,22 +78,22 @@
             return var1.acceptDouble(this.first, this.second);
          }
       }
    }
 
-   public interface Combiner<S, T> {
-      T acceptDouble(S var1, S var2);
-
-      T acceptSingle(S var1);
-
-      T acceptNone();
-   }
-
    public static enum BlockType {
       SINGLE,
       FIRST,
       SECOND;
 
       private BlockType() {
       }
    }
+
+   public interface Combiner<S, T> {
+      T acceptDouble(S var1, S var2);
+
+      T acceptSingle(S var1);
+
+      T acceptNone();
+   }
 }
