--- net/minecraft/world/level/block/LightBlock.java
+++ net/minecraft/world/level/block/LightBlock.java
@@ -24,10 +24,11 @@
 import net.minecraft.world.phys.shapes.CollisionContext;
 import net.minecraft.world.phys.shapes.Shapes;
 import net.minecraft.world.phys.shapes.VoxelShape;
 
 public class LightBlock extends Block implements SimpleWaterloggedBlock {
+   public static final int MAX_LEVEL = 15;
    public static final IntegerProperty LEVEL;
    public static final BooleanProperty WATERLOGGED;
    public static final ToIntFunction<BlockState> LIGHT_EMISSION;
 
    public LightBlock(BlockBehaviour.Properties var1) {
@@ -76,13 +77,16 @@
       return (Boolean)var1.getValue(WATERLOGGED) ? Fluids.WATER.getSource(false) : super.getFluidState(var1);
    }
 
    public ItemStack getCloneItemStack(BlockGetter var1, BlockPos var2, BlockState var3) {
       ItemStack var4 = super.getCloneItemStack(var1, var2, var3);
-      CompoundTag var5 = new CompoundTag();
-      var5.putString(LEVEL.getName(), String.valueOf(var3.getValue(LEVEL)));
-      var4.addTagElement("BlockStateTag", var5);
+      if ((Integer)var3.getValue(LEVEL) != 15) {
+         CompoundTag var5 = new CompoundTag();
+         var5.putString(LEVEL.getName(), String.valueOf(var3.getValue(LEVEL)));
+         var4.addTagElement("BlockStateTag", var5);
+      }
+
       return var4;
    }
 
    static {
       LEVEL = BlockStateProperties.LEVEL;
