--- net/minecraft/world/level/block/RedstoneTorchBlock.java
+++ net/minecraft/world/level/block/RedstoneTorchBlock.java
@@ -139,12 +139,12 @@
       LIT = BlockStateProperties.LIT;
       RECENT_TOGGLES = new WeakHashMap();
    }
 
    public static class Toggle {
-      private final BlockPos pos;
-      private final long when;
+      final BlockPos pos;
+      final long when;
 
       public Toggle(BlockPos var1, long var2) {
          this.pos = var1;
          this.when = var2;
       }
