--- net/minecraft/world/level/block/WallBlock.java
+++ net/minecraft/world/level/block/WallBlock.java
@@ -106,12 +106,12 @@
                      if (var22) {
                         var31 = Shapes.or(var31, var11);
                      }
 
                      BlockState var32 = (BlockState)((BlockState)((BlockState)((BlockState)((BlockState)this.defaultBlockState().setValue(UP, var22)).setValue(EAST_WALL, var24)).setValue(WEST_WALL, var28)).setValue(NORTH_WALL, var26)).setValue(SOUTH_WALL, var30);
-                     var20.put(var32.setValue(WATERLOGGED, false), var31);
-                     var20.put(var32.setValue(WATERLOGGED, true), var31);
+                     var20.put((BlockState)var32.setValue(WATERLOGGED, false), var31);
+                     var20.put((BlockState)var32.setValue(WATERLOGGED, true), var31);
                   }
                }
             }
          }
       }
@@ -256,26 +256,26 @@
    }
 
    public BlockState rotate(BlockState var1, Rotation var2) {
       switch(var2) {
       case CLOCKWISE_180:
-         return (BlockState)((BlockState)((BlockState)((BlockState)var1.setValue(NORTH_WALL, var1.getValue(SOUTH_WALL))).setValue(EAST_WALL, var1.getValue(WEST_WALL))).setValue(SOUTH_WALL, var1.getValue(NORTH_WALL))).setValue(WEST_WALL, var1.getValue(EAST_WALL));
+         return (BlockState)((BlockState)((BlockState)((BlockState)var1.setValue(NORTH_WALL, (WallSide)var1.getValue(SOUTH_WALL))).setValue(EAST_WALL, (WallSide)var1.getValue(WEST_WALL))).setValue(SOUTH_WALL, (WallSide)var1.getValue(NORTH_WALL))).setValue(WEST_WALL, (WallSide)var1.getValue(EAST_WALL));
       case COUNTERCLOCKWISE_90:
-         return (BlockState)((BlockState)((BlockState)((BlockState)var1.setValue(NORTH_WALL, var1.getValue(EAST_WALL))).setValue(EAST_WALL, var1.getValue(SOUTH_WALL))).setValue(SOUTH_WALL, var1.getValue(WEST_WALL))).setValue(WEST_WALL, var1.getValue(NORTH_WALL));
+         return (BlockState)((BlockState)((BlockState)((BlockState)var1.setValue(NORTH_WALL, (WallSide)var1.getValue(EAST_WALL))).setValue(EAST_WALL, (WallSide)var1.getValue(SOUTH_WALL))).setValue(SOUTH_WALL, (WallSide)var1.getValue(WEST_WALL))).setValue(WEST_WALL, (WallSide)var1.getValue(NORTH_WALL));
       case CLOCKWISE_90:
-         return (BlockState)((BlockState)((BlockState)((BlockState)var1.setValue(NORTH_WALL, var1.getValue(WEST_WALL))).setValue(EAST_WALL, var1.getValue(NORTH_WALL))).setValue(SOUTH_WALL, var1.getValue(EAST_WALL))).setValue(WEST_WALL, var1.getValue(SOUTH_WALL));
+         return (BlockState)((BlockState)((BlockState)((BlockState)var1.setValue(NORTH_WALL, (WallSide)var1.getValue(WEST_WALL))).setValue(EAST_WALL, (WallSide)var1.getValue(NORTH_WALL))).setValue(SOUTH_WALL, (WallSide)var1.getValue(EAST_WALL))).setValue(WEST_WALL, (WallSide)var1.getValue(SOUTH_WALL));
       default:
          return var1;
       }
    }
 
    public BlockState mirror(BlockState var1, Mirror var2) {
       switch(var2) {
       case LEFT_RIGHT:
-         return (BlockState)((BlockState)var1.setValue(NORTH_WALL, var1.getValue(SOUTH_WALL))).setValue(SOUTH_WALL, var1.getValue(NORTH_WALL));
+         return (BlockState)((BlockState)var1.setValue(NORTH_WALL, (WallSide)var1.getValue(SOUTH_WALL))).setValue(SOUTH_WALL, (WallSide)var1.getValue(NORTH_WALL));
       case FRONT_BACK:
-         return (BlockState)((BlockState)var1.setValue(EAST_WALL, var1.getValue(WEST_WALL))).setValue(WEST_WALL, var1.getValue(EAST_WALL));
+         return (BlockState)((BlockState)var1.setValue(EAST_WALL, (WallSide)var1.getValue(WEST_WALL))).setValue(WEST_WALL, (WallSide)var1.getValue(EAST_WALL));
       default:
          return super.mirror(var1, var2);
       }
    }
 
