--- net/minecraft/world/level/block/WeatheringCopper.java
+++ net/minecraft/world/level/block/WeatheringCopper.java
@@ -14,11 +14,11 @@
    Supplier<BiMap<Block, Block>> PREVIOUS_BY_BLOCK = Suppliers.memoize(() -> {
       return ((BiMap)NEXT_BY_BLOCK.get()).inverse();
    });
 
    static Optional<Block> getPrevious(Block var0) {
-      return Optional.ofNullable(((BiMap)PREVIOUS_BY_BLOCK.get()).get(var0));
+      return Optional.ofNullable((Block)((BiMap)PREVIOUS_BY_BLOCK.get()).get(var0));
    }
 
    static Block getFirst(Block var0) {
       Block var1 = var0;
 
@@ -34,11 +34,11 @@
          return var1.withPropertiesOf(var0);
       });
    }
 
    static Optional<Block> getNext(Block var0) {
-      return Optional.ofNullable(((BiMap)NEXT_BY_BLOCK.get()).get(var0));
+      return Optional.ofNullable((Block)((BiMap)NEXT_BY_BLOCK.get()).get(var0));
    }
 
    static BlockState getFirst(BlockState var0) {
       return getFirst(var0.getBlock()).withPropertiesOf(var0);
    }
