--- net/minecraft/advancements/critereon/BredAnimalsTrigger.java
+++ net/minecraft/advancements/critereon/BredAnimalsTrigger.java
@@ -7,11 +7,11 @@
 import net.minecraft.world.entity.AgeableMob;
 import net.minecraft.world.entity.animal.Animal;
 import net.minecraft.world.level.storage.loot.LootContext;
 
 public class BredAnimalsTrigger extends SimpleCriterionTrigger<BredAnimalsTrigger.TriggerInstance> {
-   private static final ResourceLocation ID = new ResourceLocation("bred_animals");
+   static final ResourceLocation ID = new ResourceLocation("bred_animals");
 
    public BredAnimalsTrigger() {
    }
 
    public ResourceLocation getId() {
