--- net/minecraft/advancements/critereon/MobEffectsPredicate.java
+++ net/minecraft/advancements/critereon/MobEffectsPredicate.java
@@ -103,11 +103,11 @@
          JsonObject var1 = new JsonObject();
          Iterator var2 = this.effects.entrySet().iterator();
 
          while(var2.hasNext()) {
             Entry var3 = (Entry)var2.next();
-            var1.add(Registry.MOB_EFFECT.getKey(var3.getKey()).toString(), ((MobEffectsPredicate.MobEffectInstancePredicate)var3.getValue()).serializeToJson());
+            var1.add(Registry.MOB_EFFECT.getKey((MobEffect)var3.getKey()).toString(), ((MobEffectsPredicate.MobEffectInstancePredicate)var3.getValue()).serializeToJson());
          }
 
          return var1;
       }
    }
