--- net/minecraft/advancements/critereon/UsedTotemTrigger.java
+++ net/minecraft/advancements/critereon/UsedTotemTrigger.java
@@ -5,11 +5,11 @@
 import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.ItemLike;
 
 public class UsedTotemTrigger extends SimpleCriterionTrigger<UsedTotemTrigger.TriggerInstance> {
-   private static final ResourceLocation ID = new ResourceLocation("used_totem");
+   static final ResourceLocation ID = new ResourceLocation("used_totem");
 
    public UsedTotemTrigger() {
    }
 
    public ResourceLocation getId() {
