--- net/minecraft/world/level/levelgen/feature/stateproviders/RandomizedIntStateProvider.java
+++ net/minecraft/world/level/levelgen/feature/stateproviders/RandomizedIntStateProvider.java
@@ -35,11 +35,12 @@
       this.values = var3;
       Collection var4 = var2.getPossibleValues();
 
       for(int var5 = var3.getMinValue(); var5 <= var3.getMaxValue(); ++var5) {
          if (!var4.contains(var5)) {
-            throw new IllegalArgumentException("Property value out of range: " + var2.getName() + ": " + var5);
+            String var10002 = var2.getName();
+            throw new IllegalArgumentException("Property value out of range: " + var10002 + ": " + var5);
          }
       }
 
    }
 
