--- net/minecraft/server/rcon/thread/GenericThread.java
+++ net/minecraft/server/rcon/thread/GenericThread.java
@@ -22,11 +22,12 @@
    public synchronized boolean start() {
       if (this.running) {
          return true;
       } else {
          this.running = true;
-         this.thread = new Thread(this, this.name + " #" + UNIQUE_THREAD_ID.incrementAndGet());
+         String var10004 = this.name;
+         this.thread = new Thread(this, var10004 + " #" + UNIQUE_THREAD_ID.incrementAndGet());
          this.thread.setUncaughtExceptionHandler(new DefaultUncaughtExceptionHandlerWithName(LOGGER));
          this.thread.start();
          LOGGER.info((String)"Thread {} started", (Object)this.name);
          return true;
       }
