--- net/minecraft/world/level/storage/loot/predicates/ConditionReference.java
+++ net/minecraft/world/level/storage/loot/predicates/ConditionReference.java
@@ -10,13 +10,13 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class ConditionReference implements LootItemCondition {
    private static final Logger LOGGER = LogManager.getLogger();
-   private final ResourceLocation name;
+   final ResourceLocation name;
 
-   private ConditionReference(ResourceLocation var1) {
+   ConditionReference(ResourceLocation var1) {
       this.name = var1;
    }
 
    public LootItemConditionType getType() {
       return LootItemConditions.REFERENCE;
