--- net/minecraft/client/particle/CritParticle.java
+++ net/minecraft/client/particle/CritParticle.java
@@ -3,11 +3,11 @@
 import net.minecraft.client.multiplayer.ClientLevel;
 import net.minecraft.core.particles.SimpleParticleType;
 import net.minecraft.util.Mth;
 
 public class CritParticle extends TextureSheetParticle {
-   private CritParticle(ClientLevel var1, double var2, double var4, double var6, double var8, double var10, double var12) {
+   CritParticle(ClientLevel var1, double var2, double var4, double var6, double var8, double var10, double var12) {
       super(var1, var2, var4, var6, 0.0D, 0.0D, 0.0D);
       this.friction = 0.7F;
       this.gravity = 0.5F;
       this.xd *= 0.10000000149011612D;
       this.yd *= 0.10000000149011612D;
