--- net/minecraft/client/particle/FlameParticle.java
+++ net/minecraft/client/particle/FlameParticle.java
@@ -3,11 +3,11 @@
 import net.minecraft.client.multiplayer.ClientLevel;
 import net.minecraft.core.particles.SimpleParticleType;
 import net.minecraft.util.Mth;
 
 public class FlameParticle extends RisingParticle {
-   private FlameParticle(ClientLevel var1, double var2, double var4, double var6, double var8, double var10, double var12) {
+   FlameParticle(ClientLevel var1, double var2, double var4, double var6, double var8, double var10, double var12) {
       super(var1, var2, var4, var6, var8, var10, var12);
    }
 
    public ParticleRenderType getRenderType() {
       return ParticleRenderType.PARTICLE_SHEET_OPAQUE;
