--- net/minecraft/client/particle/GlowParticle.java
+++ net/minecraft/client/particle/GlowParticle.java
@@ -4,14 +4,14 @@
 import net.minecraft.client.multiplayer.ClientLevel;
 import net.minecraft.core.particles.SimpleParticleType;
 import net.minecraft.util.Mth;
 
 public class GlowParticle extends TextureSheetParticle {
-   private static final Random RANDOM = new Random();
+   static final Random RANDOM = new Random();
    private final SpriteSet sprites;
 
-   private GlowParticle(ClientLevel var1, double var2, double var4, double var6, double var8, double var10, double var12, SpriteSet var14) {
+   GlowParticle(ClientLevel var1, double var2, double var4, double var6, double var8, double var10, double var12, SpriteSet var14) {
       super(var1, var2, var4, var6, var8, var10, var12);
       this.friction = 0.96F;
       this.speedUpWhenYMotionIsBlocked = true;
       this.sprites = var14;
       this.quadSize *= 0.75F;
