--- net/minecraft/client/particle/StationaryItemParticle.java
+++ net/minecraft/client/particle/StationaryItemParticle.java
@@ -6,11 +6,11 @@
 import net.minecraft.world.item.Items;
 import net.minecraft.world.level.ItemLike;
 import net.minecraft.world.level.block.Blocks;
 
 public class StationaryItemParticle extends TextureSheetParticle {
-   private StationaryItemParticle(ClientLevel var1, double var2, double var4, double var6, ItemLike var8) {
+   StationaryItemParticle(ClientLevel var1, double var2, double var4, double var6, ItemLike var8) {
       super(var1, var2, var4, var6);
       this.setSprite(Minecraft.getInstance().getItemRenderer().getItemModelShaper().getParticleIcon(var8));
       this.gravity = 0.0F;
       this.lifetime = 80;
       this.hasPhysics = false;
