--- net/minecraft/client/particle/VibrationSignalParticle.java
+++ net/minecraft/client/particle/VibrationSignalParticle.java
@@ -16,11 +16,11 @@
 public class VibrationSignalParticle extends TextureSheetParticle {
    private final VibrationPath vibrationPath;
    private float yRot;
    private float yRotO;
 
-   private VibrationSignalParticle(ClientLevel var1, VibrationPath var2, int var3) {
+   VibrationSignalParticle(ClientLevel var1, VibrationPath var2, int var3) {
       super(var1, (double)((float)var2.getOrigin().getX() + 0.5F), (double)((float)var2.getOrigin().getY() + 0.5F), (double)((float)var2.getOrigin().getZ() + 0.5F), 0.0D, 0.0D, 0.0D);
       this.quadSize = 0.3F;
       this.vibrationPath = var2;
       this.lifetime = var3;
    }
