--- net/minecraft/world/level/lighting/SkyLightSectionStorage.java
+++ net/minecraft/world/level/lighting/SkyLightSectionStorage.java
@@ -300,13 +300,13 @@
 
    static {
       HORIZONTALS = new Direction[]{Direction.NORTH, Direction.SOUTH, Direction.WEST, Direction.EAST};
    }
 
-   public static final class SkyDataLayerStorageMap extends DataLayerStorageMap<SkyLightSectionStorage.SkyDataLayerStorageMap> {
-      private int currentLowestY;
-      private final Long2IntOpenHashMap topSections;
+   protected static final class SkyDataLayerStorageMap extends DataLayerStorageMap<SkyLightSectionStorage.SkyDataLayerStorageMap> {
+      int currentLowestY;
+      final Long2IntOpenHashMap topSections;
 
       public SkyDataLayerStorageMap(Long2ObjectOpenHashMap<DataLayer> var1, Long2IntOpenHashMap var2, int var3) {
          super(var1);
          this.topSections = var2;
          var2.defaultReturnValue(var3);
