--- net/minecraft/client/renderer/block/model/ItemOverrides.java
+++ net/minecraft/client/renderer/block/model/ItemOverrides.java
@@ -99,18 +99,18 @@
    }
 
    static class BakedOverride {
       private final ItemOverrides.PropertyMatcher[] matchers;
       @Nullable
-      private final BakedModel model;
+      final BakedModel model;
 
-      private BakedOverride(ItemOverrides.PropertyMatcher[] var1, @Nullable BakedModel var2) {
+      BakedOverride(ItemOverrides.PropertyMatcher[] var1, @Nullable BakedModel var2) {
          this.matchers = var1;
          this.model = var2;
       }
 
-      private boolean test(float[] var1) {
+      boolean test(float[] var1) {
          ItemOverrides.PropertyMatcher[] var2 = this.matchers;
          int var3 = var2.length;
 
          for(int var4 = 0; var4 < var3; ++var4) {
             ItemOverrides.PropertyMatcher var5 = var2[var4];
@@ -122,15 +122,15 @@
 
          return true;
       }
    }
 
-   static class PropertyMatcher {
+   private static class PropertyMatcher {
       public final int index;
       public final float value;
 
-      private PropertyMatcher(int var1, float var2) {
+      PropertyMatcher(int var1, float var2) {
          this.index = var1;
          this.value = var2;
       }
    }
 }
