--- net/minecraft/client/renderer/block/model/ItemModelGenerator.java
+++ net/minecraft/client/renderer/block/model/ItemModelGenerator.java
@@ -198,11 +198,11 @@
 
    private boolean isTransparent(TextureAtlasSprite var1, int var2, int var3, int var4, int var5, int var6) {
       return var3 >= 0 && var4 >= 0 && var3 < var5 && var4 < var6 ? var1.isTransparent(var2, var3, var4) : true;
    }
 
-   static class Span {
+   private static class Span {
       private final ItemModelGenerator.SpanFacing facing;
       private int min;
       private int max;
       private final int anchor;
 
@@ -237,11 +237,11 @@
       public int getAnchor() {
          return this.anchor;
       }
    }
 
-   static enum SpanFacing {
+   private static enum SpanFacing {
       UP(Direction.UP, 0, -1),
       DOWN(Direction.DOWN, 0, 1),
       LEFT(Direction.EAST, -1, 0),
       RIGHT(Direction.WEST, 1, 0);
 
@@ -265,10 +265,10 @@
 
       public int getYOffset() {
          return this.yOffset;
       }
 
-      private boolean isHorizontal() {
+      boolean isHorizontal() {
          return this == DOWN || this == UP;
       }
    }
 }
