--- net/minecraft/server/commands/AdvancementCommands.java
+++ net/minecraft/server/commands/AdvancementCommands.java
@@ -158,27 +158,11 @@
          addChildren(var3, var1);
       }
 
    }
 
-   static enum Mode {
-      ONLY(false, false),
-      THROUGH(true, true),
-      FROM(false, true),
-      UNTIL(true, false),
-      EVERYTHING(true, true);
-
-      private final boolean parents;
-      private final boolean children;
-
-      private Mode(boolean var3, boolean var4) {
-         this.parents = var3;
-         this.children = var4;
-      }
-   }
-
-   static enum Action {
+   private static enum Action {
       GRANT("grant") {
          protected boolean perform(ServerPlayer var1, Advancement var2) {
             AdvancementProgress var3 = var1.getAdvancements().getOrStartProgress(var2);
             if (var3.isDone()) {
                return false;
@@ -220,11 +204,11 @@
          }
       };
 
       private final String key;
 
-      private Action(String var3) {
+      Action(String var3) {
          this.key = "commands.advancement." + var3;
       }
 
       public int perform(ServerPlayer var1, Iterable<Advancement> var2) {
          int var3 = 0;
@@ -246,6 +230,22 @@
 
       protected String getKey() {
          return this.key;
       }
    }
+
+   static enum Mode {
+      ONLY(false, false),
+      THROUGH(true, true),
+      FROM(false, true),
+      UNTIL(true, false),
+      EVERYTHING(true, true);
+
+      final boolean parents;
+      final boolean children;
+
+      private Mode(boolean var3, boolean var4) {
+         this.parents = var3;
+         this.children = var4;
+      }
+   }
 }
