--- net/minecraft/server/commands/CloneCommands.java
+++ net/minecraft/server/commands/CloneCommands.java
@@ -204,24 +204,11 @@
             }
          }
       }
    }
 
-   static class CloneBlockInfo {
-      public final BlockPos pos;
-      public final BlockState state;
-      @Nullable
-      public final CompoundTag tag;
-
-      public CloneBlockInfo(BlockPos var1, BlockState var2, @Nullable CompoundTag var3) {
-         this.pos = var1;
-         this.state = var2;
-         this.tag = var3;
-      }
-   }
-
-   static enum Mode {
+   private static enum Mode {
       FORCE(true),
       MOVE(true),
       NORMAL(false);
 
       private final boolean canOverlap;
@@ -232,6 +219,19 @@
 
       public boolean canOverlap() {
          return this.canOverlap;
       }
    }
+
+   static class CloneBlockInfo {
+      public final BlockPos pos;
+      public final BlockState state;
+      @Nullable
+      public final CompoundTag tag;
+
+      public CloneBlockInfo(BlockPos var1, BlockState var2, @Nullable CompoundTag var3) {
+         this.pos = var1;
+         this.state = var2;
+         this.tag = var3;
+      }
+   }
 }
