--- net/minecraft/world/level/block/CaveVinesBlock.java
+++ net/minecraft/world/level/block/CaveVinesBlock.java
@@ -35,11 +35,11 @@
    protected Block getBodyBlock() {
       return Blocks.CAVE_VINES_PLANT;
    }
 
    protected BlockState updateBodyAfterConvertedFromHead(BlockState var1, BlockState var2) {
-      return (BlockState)var2.setValue(BERRIES, var1.getValue(BERRIES));
+      return (BlockState)var2.setValue(BERRIES, (Boolean)var1.getValue(BERRIES));
    }
 
    protected BlockState getGrowIntoState(BlockState var1, Random var2) {
       return (BlockState)super.getGrowIntoState(var1, var2).setValue(BERRIES, var2.nextFloat() < 0.11F);
    }
