--- net/minecraft/world/level/block/CaveVinesPlantBlock.java
+++ net/minecraft/world/level/block/CaveVinesPlantBlock.java
@@ -25,11 +25,11 @@
    protected GrowingPlantHeadBlock getHeadBlock() {
       return (GrowingPlantHeadBlock)Blocks.CAVE_VINES;
    }
 
    protected BlockState updateHeadAfterConvertedFromBody(BlockState var1, BlockState var2) {
-      return (BlockState)var2.setValue(BERRIES, var1.getValue(BERRIES));
+      return (BlockState)var2.setValue(BERRIES, (Boolean)var1.getValue(BERRIES));
    }
 
    public ItemStack getCloneItemStack(BlockGetter var1, BlockPos var2, BlockState var3) {
       return new ItemStack(Items.GLOW_BERRIES);
    }
