--- net/minecraft/world/entity/animal/horse/TraderLlama.java
+++ net/minecraft/world/entity/animal/horse/TraderLlama.java
@@ -106,11 +106,11 @@
       }
 
       return super.finalizeSpawn(var1, var2, var3, (SpawnGroupData)var4, var5);
    }
 
-   public static class TraderLlamaDefendWanderingTraderGoal extends TargetGoal {
+   protected static class TraderLlamaDefendWanderingTraderGoal extends TargetGoal {
       private final Llama llama;
       private LivingEntity ownerLastHurtBy;
       private int timestamp;
 
       public TraderLlamaDefendWanderingTraderGoal(Llama var1) {
