--- net/minecraft/world/level/storage/loot/entries/TagEntry.java
+++ net/minecraft/world/level/storage/loot/entries/TagEntry.java
@@ -16,14 +16,14 @@
 import net.minecraft.world.level.storage.loot.LootContext;
 import net.minecraft.world.level.storage.loot.functions.LootItemFunction;
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
 
 public class TagEntry extends LootPoolSingletonContainer {
-   private final Tag<Item> tag;
-   private final boolean expand;
+   final Tag<Item> tag;
+   final boolean expand;
 
-   private TagEntry(Tag<Item> var1, boolean var2, int var3, int var4, LootItemCondition[] var5, LootItemFunction[] var6) {
+   TagEntry(Tag<Item> var1, boolean var2, int var3, int var4, LootItemCondition[] var5, LootItemFunction[] var6) {
       super(var3, var4, var5, var6);
       this.tag = var1;
       this.expand = var2;
    }
 
