--- net/minecraft/advancements/critereon/ConsumeItemTrigger.java
+++ net/minecraft/advancements/critereon/ConsumeItemTrigger.java
@@ -7,11 +7,11 @@
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.alchemy.Potion;
 import net.minecraft.world.level.ItemLike;
 
 public class ConsumeItemTrigger extends SimpleCriterionTrigger<ConsumeItemTrigger.TriggerInstance> {
-   private static final ResourceLocation ID = new ResourceLocation("consume_item");
+   static final ResourceLocation ID = new ResourceLocation("consume_item");
 
    public ConsumeItemTrigger() {
    }
 
    public ResourceLocation getId() {
