--- net/minecraft/advancements/critereon/EntityHurtPlayerTrigger.java
+++ net/minecraft/advancements/critereon/EntityHurtPlayerTrigger.java
@@ -4,11 +4,11 @@
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.world.damagesource.DamageSource;
 
 public class EntityHurtPlayerTrigger extends SimpleCriterionTrigger<EntityHurtPlayerTrigger.TriggerInstance> {
-   private static final ResourceLocation ID = new ResourceLocation("entity_hurt_player");
+   static final ResourceLocation ID = new ResourceLocation("entity_hurt_player");
 
    public EntityHurtPlayerTrigger() {
    }
 
    public ResourceLocation getId() {
