--- net/minecraft/advancements/critereon/KilledTrigger.java
+++ net/minecraft/advancements/critereon/KilledTrigger.java
@@ -7,11 +7,11 @@
 import net.minecraft.world.damagesource.DamageSource;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.level.storage.loot.LootContext;
 
 public class KilledTrigger extends SimpleCriterionTrigger<KilledTrigger.TriggerInstance> {
-   private final ResourceLocation id;
+   final ResourceLocation id;
 
    public KilledTrigger(ResourceLocation var1) {
       this.id = var1;
    }
 
