--- net/minecraft/advancements/critereon/LightPredicate.java
+++ net/minecraft/advancements/critereon/LightPredicate.java
@@ -10,11 +10,11 @@
 
 public class LightPredicate {
    public static final LightPredicate ANY;
    private final MinMaxBounds.Ints composite;
 
-   private LightPredicate(MinMaxBounds.Ints var1) {
+   LightPredicate(MinMaxBounds.Ints var1) {
       this.composite = var1;
    }
 
    public boolean matches(ServerLevel var1, BlockPos var2) {
       if (this == ANY) {
