--- net/minecraft/advancements/critereon/LocationTrigger.java
+++ net/minecraft/advancements/critereon/LocationTrigger.java
@@ -6,11 +6,11 @@
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.util.GsonHelper;
 
 public class LocationTrigger extends SimpleCriterionTrigger<LocationTrigger.TriggerInstance> {
-   private final ResourceLocation id;
+   final ResourceLocation id;
 
    public LocationTrigger(ResourceLocation var1) {
       this.id = var1;
    }
 
