--- net/minecraft/advancements/critereon/PlayerHurtEntityTrigger.java
+++ net/minecraft/advancements/critereon/PlayerHurtEntityTrigger.java
@@ -6,11 +6,11 @@
 import net.minecraft.world.damagesource.DamageSource;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.level.storage.loot.LootContext;
 
 public class PlayerHurtEntityTrigger extends SimpleCriterionTrigger<PlayerHurtEntityTrigger.TriggerInstance> {
-   private static final ResourceLocation ID = new ResourceLocation("player_hurt_entity");
+   static final ResourceLocation ID = new ResourceLocation("player_hurt_entity");
 
    public PlayerHurtEntityTrigger() {
    }
 
    public ResourceLocation getId() {
