--- net/minecraft/client/gui/components/toasts/AdvancementToast.java
+++ net/minecraft/client/gui/components/toasts/AdvancementToast.java
@@ -2,10 +2,11 @@
 
 import com.mojang.blaze3d.systems.RenderSystem;
 import com.mojang.blaze3d.vertex.PoseStack;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Objects;
 import net.minecraft.advancements.Advancement;
 import net.minecraft.advancements.DisplayInfo;
 import net.minecraft.advancements.FrameType;
 import net.minecraft.client.renderer.GameRenderer;
 import net.minecraft.client.resources.sounds.SimpleSoundInstance;
@@ -42,17 +43,17 @@
                var2.getMinecraft().font.draw(var1, var5.getFrame().getDisplayName(), 30.0F, 11.0F, var7 | var10);
             } else {
                var10 = Mth.floor(Mth.clamp((float)(var3 - 1500L) / 300.0F, 0.0F, 1.0F) * 252.0F) << 24 | 67108864;
                int var10000 = this.height() / 2;
                int var10001 = var6.size();
-               var2.getMinecraft().font.getClass();
+               Objects.requireNonNull(var2.getMinecraft().font);
                int var11 = var10000 - var10001 * 9 / 2;
 
                for(Iterator var12 = var6.iterator(); var12.hasNext(); var11 += 9) {
                   FormattedCharSequence var13 = (FormattedCharSequence)var12.next();
                   var2.getMinecraft().font.draw(var1, var13, 30.0F, (float)var11, 16777215 | var10);
-                  var2.getMinecraft().font.getClass();
+                  Objects.requireNonNull(var2.getMinecraft().font);
                }
             }
          }
 
          if (!this.playedSound && var3 > 0L) {
