--- net/minecraft/client/renderer/CubeMap.java
+++ net/minecraft/client/renderer/CubeMap.java
@@ -18,11 +18,14 @@
    private static final int SIDES = 6;
    private final ResourceLocation[] images = new ResourceLocation[6];
 
    public CubeMap(ResourceLocation var1) {
       for(int var2 = 0; var2 < 6; ++var2) {
-         this.images[var2] = new ResourceLocation(var1.getNamespace(), var1.getPath() + '_' + var2 + ".png");
+         ResourceLocation[] var10000 = this.images;
+         String var10004 = var1.getNamespace();
+         String var10005 = var1.getPath();
+         var10000[var2] = new ResourceLocation(var10004, var10005 + "_" + var2 + ".png");
       }
 
    }
 
    public void render(Minecraft var1, float var2, float var3, float var4) {
