--- net/minecraft/client/renderer/FaceInfo.java
+++ net/minecraft/client/renderer/FaceInfo.java
@@ -36,11 +36,11 @@
    public static class VertexInfo {
       public final int xFace;
       public final int yFace;
       public final int zFace;
 
-      private VertexInfo(int var1, int var2, int var3) {
+      VertexInfo(int var1, int var2, int var3) {
          this.xFace = var1;
          this.yFace = var2;
          this.zFace = var3;
       }
    }
