--- net/minecraft/client/renderer/OutlineBufferSource.java
+++ net/minecraft/client/renderer/OutlineBufferSource.java
@@ -53,11 +53,11 @@
       private double y;
       private double z;
       private float u;
       private float v;
 
-      private EntityOutlineGenerator(VertexConsumer var1, int var2, int var3, int var4, int var5) {
+      EntityOutlineGenerator(VertexConsumer var1, int var2, int var3, int var4, int var5) {
          this.delegate = var1;
          super.defaultColor(var2, var3, var4, var5);
       }
 
       public void defaultColor(int var1, int var2, int var3, int var4) {
