--- net/minecraft/world/level/storage/loot/providers/number/ScoreboardValue.java
+++ net/minecraft/world/level/storage/loot/providers/number/ScoreboardValue.java
@@ -11,15 +11,15 @@
 import net.minecraft.world.level.storage.loot.providers.score.ContextScoreboardNameProvider;
 import net.minecraft.world.level.storage.loot.providers.score.ScoreboardNameProvider;
 import net.minecraft.world.scores.Objective;
 
 public class ScoreboardValue implements NumberProvider {
-   private final ScoreboardNameProvider target;
-   private final String score;
-   private final float scale;
+   final ScoreboardNameProvider target;
+   final String score;
+   final float scale;
 
-   private ScoreboardValue(ScoreboardNameProvider var1, String var2, float var3) {
+   ScoreboardValue(ScoreboardNameProvider var1, String var2, float var3) {
       this.target = var1;
       this.score = var2;
       this.scale = var3;
    }
 
