--- net/minecraft/world/item/crafting/ShapelessRecipe.java
+++ net/minecraft/world/item/crafting/ShapelessRecipe.java
@@ -14,13 +14,13 @@
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.Level;
 
 public class ShapelessRecipe implements CraftingRecipe {
    private final ResourceLocation id;
-   private final String group;
-   private final ItemStack result;
-   private final NonNullList<Ingredient> ingredients;
+   final String group;
+   final ItemStack result;
+   final NonNullList<Ingredient> ingredients;
 
    public ShapelessRecipe(ResourceLocation var1, String var2, ItemStack var3, NonNullList<Ingredient> var4) {
       this.id = var1;
       this.group = var2;
       this.result = var3;
