--- net/minecraft/world/entity/schedule/Timeline.java
+++ net/minecraft/world/entity/schedule/Timeline.java
@@ -30,11 +30,11 @@
    }
 
    private void sortAndDeduplicateKeyframes() {
       Int2ObjectAVLTreeMap var1 = new Int2ObjectAVLTreeMap();
       this.keyframes.forEach((var1x) -> {
-         Keyframe var10000 = (Keyframe)var1.put(var1x.getTimeStamp(), var1x);
+         var1.put(var1x.getTimeStamp(), var1x);
       });
       this.keyframes.clear();
       this.keyframes.addAll(var1.values());
       this.previousIndex = 0;
    }
