--- net/minecraft/util/WeighedRandom.java
+++ net/minecraft/util/WeighedRandom.java
@@ -8,11 +8,11 @@
 import net.minecraft.Util;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class WeighedRandom {
-   private static final Logger LOGGER = LogManager.getLogger();
+   static final Logger LOGGER = LogManager.getLogger();
 
    public WeighedRandom() {
    }
 
    public static int getTotalWeight(List<? extends WeighedRandom.WeighedRandomItem> var0) {
