--- net/minecraft/world/item/CompassItem.java
+++ net/minecraft/world/item/CompassItem.java
@@ -1,8 +1,9 @@
 package net.minecraft.world.item;
 
 import com.mojang.serialization.DataResult;
+import java.util.Objects;
 import java.util.Optional;
 import net.minecraft.core.BlockPos;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.nbt.NbtOps;
 import net.minecraft.nbt.NbtUtils;
@@ -95,11 +96,11 @@
 
    private void addLodestoneTags(ResourceKey<Level> var1, BlockPos var2, CompoundTag var3) {
       var3.put("LodestonePos", NbtUtils.writeBlockPos(var2));
       DataResult var10000 = Level.RESOURCE_KEY_CODEC.encodeStart(NbtOps.INSTANCE, var1);
       Logger var10001 = LOGGER;
-      var10001.getClass();
+      Objects.requireNonNull(var10001);
       var10000.resultOrPartial(var10001::error).ifPresent((var1x) -> {
          var3.put("LodestoneDimension", var1x);
       });
       var3.putBoolean("LodestoneTracked", true);
    }
