--- net/minecraft/world/item/ItemStack.java
+++ net/minecraft/world/item/ItemStack.java
@@ -450,11 +450,12 @@
    public String getDescriptionId() {
       return this.getItem().getDescriptionId(this);
    }
 
    public String toString() {
-      return this.count + " " + this.getItem();
+      int var10000 = this.count;
+      return var10000 + " " + this.getItem();
    }
 
    public void inventoryTick(Level var1, Entity var2, int var3, boolean var4) {
       if (this.popTime > 0) {
          --this.popTime;
@@ -872,11 +873,11 @@
             if (!var5.contains("Slot", 8) || var5.getString("Slot").equals(var1.getName())) {
                Optional var6 = Registry.ATTRIBUTE.getOptional(ResourceLocation.tryParse(var5.getString("AttributeName")));
                if (var6.isPresent()) {
                   AttributeModifier var7 = AttributeModifier.load(var5);
                   if (var7 != null && var7.getId().getLeastSignificantBits() != 0L && var7.getId().getMostSignificantBits() != 0L) {
-                     ((Multimap)var2).put(var6.get(), var7);
+                     ((Multimap)var2).put((Attribute)var6.get(), var7);
                   }
                }
             }
          }
       } else {
