--- net/minecraft/world/item/ShieldItem.java
+++ net/minecraft/world/item/ShieldItem.java
@@ -20,11 +20,16 @@
       super(var1);
       DispenserBlock.registerBehavior(this, ArmorItem.DISPENSE_ITEM_BEHAVIOR);
    }
 
    public String getDescriptionId(ItemStack var1) {
-      return var1.getTagElement("BlockEntityTag") != null ? this.getDescriptionId() + '.' + getColor(var1).getName() : super.getDescriptionId(var1);
+      if (var1.getTagElement("BlockEntityTag") != null) {
+         String var10000 = this.getDescriptionId();
+         return var10000 + "." + getColor(var1).getName();
+      } else {
+         return super.getDescriptionId(var1);
+      }
    }
 
    public void appendHoverText(ItemStack var1, @Nullable Level var2, List<Component> var3, TooltipFlag var4) {
       BannerItem.appendHoverTextFromBannerBlockEntityTag(var1, var3);
    }
