--- net/minecraft/world/entity/ai/goal/target/TargetGoal.java
+++ net/minecraft/world/entity/ai/goal/target/TargetGoal.java
@@ -54,18 +54,18 @@
             double var4 = this.getFollowDistance();
             if (this.mob.distanceToSqr(var1) > var4 * var4) {
                return false;
             } else {
                if (this.mustSee) {
-                  if (this.mob.getSensing().canSee(var1)) {
+                  if (this.mob.getSensing().hasLineOfSight(var1)) {
                      this.unseenTicks = 0;
                   } else if (++this.unseenTicks > this.unseenMemoryTicks) {
                      return false;
                   }
                }
 
-               if (!var1.canBeTargeted()) {
+               if (!var1.canBeSeenAsEnemy()) {
                   return false;
                } else {
                   this.mob.setTarget(var1);
                   return true;
                }
