--- net/minecraft/world/level/storage/loot/predicates/ValueCheckCondition.java
+++ net/minecraft/world/level/storage/loot/predicates/ValueCheckCondition.java
@@ -10,14 +10,14 @@
 import net.minecraft.world.level.storage.loot.LootContext;
 import net.minecraft.world.level.storage.loot.parameters.LootContextParam;
 import net.minecraft.world.level.storage.loot.providers.number.NumberProvider;
 
 public class ValueCheckCondition implements LootItemCondition {
-   private final NumberProvider provider;
-   private final IntRange range;
+   final NumberProvider provider;
+   final IntRange range;
 
-   private ValueCheckCondition(NumberProvider var1, IntRange var2) {
+   ValueCheckCondition(NumberProvider var1, IntRange var2) {
       this.provider = var1;
       this.range = var2;
    }
 
    public LootItemConditionType getType() {
