--- net/minecraft/network/protocol/game/ClientboundGameEventPacket.java
+++ net/minecraft/network/protocol/game/ClientboundGameEventPacket.java
@@ -52,12 +52,12 @@
    public float getParam() {
       return this.param;
    }
 
    public static class Type {
-      private static final Int2ObjectMap<ClientboundGameEventPacket.Type> TYPES = new Int2ObjectOpenHashMap();
-      private final int id;
+      static final Int2ObjectMap<ClientboundGameEventPacket.Type> TYPES = new Int2ObjectOpenHashMap();
+      final int id;
 
       public Type(int var1) {
          this.id = var1;
          TYPES.put(var1, this);
       }
