--- net/minecraft/network/protocol/game/DebugEntityNameGenerator.java
+++ net/minecraft/network/protocol/game/DebugEntityNameGenerator.java
@@ -23,11 +23,12 @@
       }
    }
 
    public static String getEntityName(UUID var0) {
       Random var1 = getRandom(var0);
-      return getRandomString(var1, NAMES_FIRST_PART) + getRandomString(var1, NAMES_SECOND_PART);
+      String var10000 = getRandomString(var1, NAMES_FIRST_PART);
+      return var10000 + getRandomString(var1, NAMES_SECOND_PART);
    }
 
    private static String getRandomString(Random var0, String[] var1) {
       return (String)Util.getRandom((Object[])var1, var0);
    }
