--- net/minecraft/network/protocol/game/ServerGamePacketListener.java
+++ net/minecraft/network/protocol/game/ServerGamePacketListener.java
@@ -25,10 +25,12 @@
 
    void handleKeepAlive(ServerboundKeepAlivePacket var1);
 
    void handleMovePlayer(ServerboundMovePlayerPacket var1);
 
+   void handlePong(ServerboundPongPacket var1);
+
    void handlePlayerAbilities(ServerboundPlayerAbilitiesPacket var1);
 
    void handlePlayerAction(ServerboundPlayerActionPacket var1);
 
    void handlePlayerCommand(ServerboundPlayerCommandPacket var1);
