--- net/minecraft/world/phys/shapes/DiscreteVoxelShape.java
+++ net/minecraft/world/phys/shapes/DiscreteVoxelShape.java
@@ -204,13 +204,13 @@
          }
       }
 
    }
 
-   public interface IntFaceConsumer {
-      void consume(Direction var1, int var2, int var3, int var4);
-   }
-
    public interface IntLineConsumer {
       void consume(int var1, int var2, int var3, int var4, int var5, int var6);
    }
+
+   public interface IntFaceConsumer {
+      void consume(Direction var1, int var2, int var3, int var4);
+   }
 }
