--- net/minecraft/world/level/levelgen/Cavifier.java
+++ net/minecraft/world/level/levelgen/Cavifier.java
@@ -137,15 +137,15 @@
 
    private static double sampleWithRarity(NormalNoise var0, double var1, double var3, double var5, double var7) {
       return var0.getValue(var1 / var7, var3 / var7, var5 / var7);
    }
 
-   static final class QuantizedSpaghettiRarity {
+   private static final class QuantizedSpaghettiRarity {
       private QuantizedSpaghettiRarity() {
       }
 
-      private static double getSphaghettiRarity2D(double var0) {
+      static double getSphaghettiRarity2D(double var0) {
          if (var0 < -0.75D) {
             return 0.5D;
          } else if (var0 < -0.5D) {
             return 0.75D;
          } else if (var0 < 0.5D) {
@@ -153,11 +153,11 @@
          } else {
             return var0 < 0.75D ? 2.0D : 3.0D;
          }
       }
 
-      private static double getSpaghettiRarity3D(double var0) {
+      static double getSpaghettiRarity3D(double var0) {
          if (var0 < -0.5D) {
             return 0.75D;
          } else if (var0 < 0.0D) {
             return 1.0D;
          } else {
