--- net/minecraft/client/renderer/entity/layers/VillagerProfessionLayer.java
+++ net/minecraft/client/renderer/entity/layers/VillagerProfessionLayer.java
@@ -73,40 +73,44 @@
 
    public <K> VillagerMetaDataSection.Hat getHatData(Object2ObjectMap<K, VillagerMetaDataSection.Hat> var1, String var2, DefaultedRegistry<K> var3, K var4) {
       return (VillagerMetaDataSection.Hat)var1.computeIfAbsent(var4, (var4x) -> {
          try {
             Resource var5 = this.resourceManager.getResource(this.getResourceLocation(var2, var3.getKey(var4)));
-            Throwable var6 = null;
 
-            VillagerMetaDataSection.Hat var8;
-            try {
-               VillagerMetaDataSection var7 = (VillagerMetaDataSection)var5.getMetadata(VillagerMetaDataSection.SERIALIZER);
-               if (var7 == null) {
-                  return VillagerMetaDataSection.Hat.NONE;
-               }
-
-               var8 = var7.getHat();
-            } catch (Throwable var19) {
-               var6 = var19;
-               throw var19;
-            } finally {
-               if (var5 != null) {
+            VillagerMetaDataSection.Hat var7;
+            label50: {
+               try {
+                  VillagerMetaDataSection var6 = (VillagerMetaDataSection)var5.getMetadata(VillagerMetaDataSection.SERIALIZER);
                   if (var6 != null) {
+                     var7 = var6.getHat();
+                     break label50;
+                  }
+               } catch (Throwable var9) {
+                  if (var5 != null) {
                      try {
                         var5.close();
-                     } catch (Throwable var18) {
-                        var6.addSuppressed(var18);
+                     } catch (Throwable var8) {
+                        var9.addSuppressed(var8);
                      }
-                  } else {
-                     var5.close();
                   }
+
+                  throw var9;
                }
 
+               if (var5 != null) {
+                  var5.close();
+               }
+
+               return VillagerMetaDataSection.Hat.NONE;
             }
 
-            return var8;
-         } catch (IOException var21) {
+            if (var5 != null) {
+               var5.close();
+            }
+
+            return var7;
+         } catch (IOException var10) {
             return VillagerMetaDataSection.Hat.NONE;
          }
       });
    }
 }
