--- net/minecraft/util/datafix/fixes/BlockRenameFixWithJigsaw.java
+++ net/minecraft/util/datafix/fixes/BlockRenameFixWithJigsaw.java
@@ -6,10 +6,11 @@
 import com.mojang.datafixers.OpticFinder;
 import com.mojang.datafixers.TypeRewriteRule;
 import com.mojang.datafixers.DSL.TypeReference;
 import com.mojang.datafixers.schemas.Schema;
 import com.mojang.serialization.Dynamic;
+import java.util.Objects;
 import java.util.Optional;
 import java.util.function.Function;
 
 public abstract class BlockRenameFixWithJigsaw extends BlockRenameFix {
    private final String name;
@@ -41,11 +42,11 @@
 
                      String var5 = var1x.substring(0, var4);
                      String var6 = this.fixBlock(var5);
                      return var6 + var1x.substring(var4);
                   });
-                  var1.getClass();
+                  Objects.requireNonNull(var1);
                   return (Dynamic)DataFixUtils.orElse(var10000.map(var1::createString), var2);
                });
             });
          });
       });
