--- net/minecraft/server/level/ThreadedLevelLightEngine.java
+++ net/minecraft/server/level/ThreadedLevelLightEngine.java
@@ -75,11 +75,11 @@
          for(var2 = this.levelHeightAccessor.getMinSection(); var2 < this.levelHeightAccessor.getMaxSection(); ++var2) {
             super.updateSectionStatus(SectionPos.of(var1, var2), true);
          }
 
       }, () -> {
-         return "updateChunkStatus " + var1 + " " + true;
+         return "updateChunkStatus " + var1 + " true";
       }));
    }
 
    public void updateSectionStatus(SectionPos var1, boolean var2) {
       this.addTask(var1.x(), var1.z(), () -> {
