--- net/minecraft/world/item/alchemy/PotionUtils.java
+++ net/minecraft/world/item/alchemy/PotionUtils.java
@@ -180,11 +180,11 @@
 
                while(var10.hasNext()) {
                   Entry var11 = (Entry)var10.next();
                   AttributeModifier var12 = (AttributeModifier)var11.getValue();
                   AttributeModifier var13 = new AttributeModifier(var12.getName(), var8.getAttributeModifierValue(var6.getAmplifier(), var12), var12.getOperation());
-                  var4.add(new Pair(var11.getKey(), var13));
+                  var4.add(new Pair((Attribute)var11.getKey(), var13));
                }
             }
 
             if (var6.getAmplifier() > 0) {
                var7 = new TranslatableComponent("potion.withAmplifier", new Object[]{var7, new TranslatableComponent("potion.potency." + var6.getAmplifier())});
