--- net/minecraft/util/valueproviders/TrapezoidFloat.java
+++ net/minecraft/util/valueproviders/TrapezoidFloat.java
@@ -55,8 +55,8 @@
    public FloatProviderType<?> getType() {
       return FloatProviderType.TRAPEZOID;
    }
 
    public String toString() {
-      return "trapezoid(" + this.plateau + ") in [" + this.min + '-' + this.max + ']';
+      return "trapezoid(" + this.plateau + ") in [" + this.min + "-" + this.max + "]";
    }
 }
