--- net/minecraft/world/entity/ai/goal/target/NearestAttackableTargetGoal.java
+++ net/minecraft/world/entity/ai/goal/target/NearestAttackableTargetGoal.java
@@ -28,11 +28,11 @@
    public NearestAttackableTargetGoal(Mob var1, Class<T> var2, int var3, boolean var4, boolean var5, @Nullable Predicate<LivingEntity> var6) {
       super(var1, var4, var5);
       this.targetType = var2;
       this.randomInterval = var3;
       this.setFlags(EnumSet.of(Goal.Flag.TARGET));
-      this.targetConditions = (new TargetingConditions()).range(this.getFollowDistance()).selector(var6);
+      this.targetConditions = TargetingConditions.forCombat().range(this.getFollowDistance()).selector(var6);
    }
 
    public boolean canUse() {
       if (this.randomInterval > 0 && this.mob.getRandom().nextInt(this.randomInterval) != 0) {
          return false;
