--- net/minecraft/world/entity/ai/goal/RangedAttackGoal.java
+++ net/minecraft/world/entity/ai/goal/RangedAttackGoal.java
@@ -59,11 +59,11 @@
       this.attackTime = -1;
    }
 
    public void tick() {
       double var1 = this.mob.distanceToSqr(this.target.getX(), this.target.getY(), this.target.getZ());
-      boolean var3 = this.mob.getSensing().canSee(this.target);
+      boolean var3 = this.mob.getSensing().hasLineOfSight(this.target);
       if (var3) {
          ++this.seeTime;
       } else {
          this.seeTime = 0;
       }
