--- net/minecraft/world/level/block/state/properties/Property.java
+++ net/minecraft/world/level/block/state/properties/Property.java
@@ -97,11 +97,11 @@
 
    public static final class Value<T extends Comparable<T>> {
       private final Property<T> property;
       private final T value;
 
-      private Value(Property<T> var1, T var2) {
+      Value(Property<T> var1, T var2) {
          if (!var1.getPossibleValues().contains(var2)) {
             throw new IllegalArgumentException("Value " + var2 + " does not belong to property " + var1);
          } else {
             this.property = var1;
             this.value = var2;
@@ -115,11 +115,12 @@
       public T value() {
          return this.value;
       }
 
       public String toString() {
-         return this.property.getName() + "=" + this.property.getName(this.value);
+         String var10000 = this.property.getName();
+         return var10000 + "=" + this.property.getName(this.value);
       }
 
       public boolean equals(Object var1) {
          if (this == var1) {
             return true;
