--- net/minecraft/world/level/biome/FuzzyOffsetBiomeZoomer.java
+++ net/minecraft/world/level/biome/FuzzyOffsetBiomeZoomer.java
@@ -63,11 +63,11 @@
       double var17 = getFiddle(var11);
       return sqr(var9 + var17) + sqr(var7 + var15) + sqr(var5 + var13);
    }
 
    private static double getFiddle(long var0) {
-      double var2 = (double)((int)Math.floorMod(var0 >> 24, 1024L)) / 1024.0D;
+      double var2 = (double)Math.floorMod(var0 >> 24, 1024) / 1024.0D;
       return (var2 - 0.5D) * 0.9D;
    }
 
    private static double sqr(double var0) {
       return var0 * var0;
