--- net/minecraft/world/entity/monster/Strider.java
+++ net/minecraft/world/entity/monster/Strider.java
@@ -444,11 +444,11 @@
    }
 
    static class StriderGoToLavaGoal extends MoveToBlockGoal {
       private final Strider strider;
 
-      private StriderGoToLavaGoal(Strider var1, double var2) {
+      StriderGoToLavaGoal(Strider var1, double var2) {
          super(var1, var2, 8, 2);
          this.strider = var1;
       }
 
       public BlockPos getMoveToTarget() {
