--- net/minecraft/world/entity/boss/enderdragon/phases/DragonLandingApproachPhase.java
+++ net/minecraft/world/entity/boss/enderdragon/phases/DragonLandingApproachPhase.java
@@ -10,11 +10,11 @@
 import net.minecraft.world.level.pathfinder.Node;
 import net.minecraft.world.level.pathfinder.Path;
 import net.minecraft.world.phys.Vec3;
 
 public class DragonLandingApproachPhase extends AbstractDragonPhaseInstance {
-   private static final TargetingConditions NEAR_EGG_TARGETING = (new TargetingConditions()).range(128.0D);
+   private static final TargetingConditions NEAR_EGG_TARGETING = TargetingConditions.forCombat().range(128.0D);
    private Path currentPath;
    private Vec3 targetLocation;
 
    public DragonLandingApproachPhase(EnderDragon var1) {
       super(var1);
