--- net/minecraft/world/entity/boss/enderdragon/phases/DragonSittingScanningPhase.java
+++ net/minecraft/world/entity/boss/enderdragon/phases/DragonSittingScanningPhase.java
@@ -9,17 +9,17 @@
 public class DragonSittingScanningPhase extends AbstractDragonSittingPhase {
    private static final int SITTING_SCANNING_IDLE_TICKS = 100;
    private static final int SITTING_ATTACK_Y_VIEW_RANGE = 10;
    private static final int SITTING_ATTACK_VIEW_RANGE = 20;
    private static final int SITTING_CHARGE_VIEW_RANGE = 150;
-   private static final TargetingConditions CHARGE_TARGETING = (new TargetingConditions()).range(150.0D);
+   private static final TargetingConditions CHARGE_TARGETING = TargetingConditions.forCombat().range(150.0D);
    private final TargetingConditions scanTargeting;
    private int scanningTime;
 
    public DragonSittingScanningPhase(EnderDragon var1) {
       super(var1);
-      this.scanTargeting = (new TargetingConditions()).range(20.0D).selector((var1x) -> {
+      this.scanTargeting = TargetingConditions.forCombat().range(20.0D).selector((var1x) -> {
          return Math.abs(var1x.getY() - var1.getY()) <= 10.0D;
       });
    }
 
    public void doServerTick() {
