--- net/minecraft/world/level/levelgen/structure/IglooPieces.java
+++ net/minecraft/world/level/levelgen/structure/IglooPieces.java
@@ -27,15 +27,15 @@
 import net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate;
 import net.minecraft.world.level.storage.loot.BuiltInLootTables;
 
 public class IglooPieces {
    public static final int GENERATION_HEIGHT = 90;
-   private static final ResourceLocation STRUCTURE_LOCATION_IGLOO = new ResourceLocation("igloo/top");
+   static final ResourceLocation STRUCTURE_LOCATION_IGLOO = new ResourceLocation("igloo/top");
    private static final ResourceLocation STRUCTURE_LOCATION_LADDER = new ResourceLocation("igloo/middle");
    private static final ResourceLocation STRUCTURE_LOCATION_LABORATORY = new ResourceLocation("igloo/bottom");
-   private static final Map<ResourceLocation, BlockPos> PIVOTS;
-   private static final Map<ResourceLocation, BlockPos> OFFSETS;
+   static final Map<ResourceLocation, BlockPos> PIVOTS;
+   static final Map<ResourceLocation, BlockPos> OFFSETS;
 
    public IglooPieces() {
    }
 
    public static void addPieces(StructureManager var0, BlockPos var1, Rotation var2, StructurePieceAccessor var3, Random var4) {
