--- net/minecraft/client/multiplayer/PlayerInfo.java
+++ net/minecraft/client/multiplayer/PlayerInfo.java
@@ -70,11 +70,11 @@
       return this.skinModel == null ? DefaultPlayerSkin.getSkinModelName(this.profile.getId()) : this.skinModel;
    }
 
    public ResourceLocation getSkinLocation() {
       this.registerTextures();
-      return (ResourceLocation)MoreObjects.firstNonNull(this.textureLocations.get(Type.SKIN), DefaultPlayerSkin.getDefaultSkin(this.profile.getId()));
+      return (ResourceLocation)MoreObjects.firstNonNull((ResourceLocation)this.textureLocations.get(Type.SKIN), DefaultPlayerSkin.getDefaultSkin(this.profile.getId()));
    }
 
    @Nullable
    public ResourceLocation getCapeLocation() {
       this.registerTextures();
