--- net/minecraft/commands/arguments/AngleArgument.java
+++ net/minecraft/commands/arguments/AngleArgument.java
@@ -48,11 +48,11 @@
 
    public static final class SingleAngle {
       private final float angle;
       private final boolean isRelative;
 
-      private SingleAngle(float var1, boolean var2) {
+      SingleAngle(float var1, boolean var2) {
          this.angle = var1;
          this.isRelative = var2;
       }
 
       public float getAngle(CommandSourceStack var1) {
