--- net/minecraft/CrashReportCategory.java
+++ net/minecraft/CrashReportCategory.java
@@ -2,10 +2,11 @@
 
 import com.google.common.collect.Lists;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Locale;
+import java.util.Objects;
 import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.SectionPos;
 import net.minecraft.world.level.LevelHeightAccessor;
 import net.minecraft.world.level.block.state.BlockState;
@@ -180,29 +181,31 @@
       return this.stackTrace;
    }
 
    public static void populateBlockDetails(CrashReportCategory var0, LevelHeightAccessor var1, BlockPos var2, @Nullable BlockState var3) {
       if (var3 != null) {
+         Objects.requireNonNull(var3);
          var0.setDetail("Block", var3::toString);
       }
 
       var0.setDetail("Block location", () -> {
          return formatLocation(var1, var2);
       });
    }
 
-   static class Entry {
+   private static class Entry {
       private final String key;
       private final String value;
 
       public Entry(String var1, @Nullable Object var2) {
          this.key = var1;
          if (var2 == null) {
             this.value = "~~NULL~~";
          } else if (var2 instanceof Throwable) {
             Throwable var3 = (Throwable)var2;
-            this.value = "~~ERROR~~ " + var3.getClass().getSimpleName() + ": " + var3.getMessage();
+            String var10001 = var3.getClass().getSimpleName();
+            this.value = "~~ERROR~~ " + var10001 + ": " + var3.getMessage();
          } else {
             this.value = var2.toString();
          }
 
       }
