--- com/mojang/realmsclient/dto/RealmsServer.java
+++ com/mojang/realmsclient/dto/RealmsServer.java
@@ -280,11 +280,11 @@
       HashMap var2 = Maps.newHashMap();
       Iterator var3 = var1.entrySet().iterator();
 
       while(var3.hasNext()) {
          Entry var4 = (Entry)var3.next();
-         var2.put(var4.getKey(), ((RealmsWorldOptions)var4.getValue()).clone());
+         var2.put((Integer)var4.getKey(), ((RealmsWorldOptions)var4.getValue()).clone());
       }
 
       return var2;
    }
 
@@ -294,10 +294,19 @@
 
    public ServerData toServerData(String var1) {
       return new ServerData(this.name, var1, false);
    }
 
+   public static enum State {
+      CLOSED,
+      OPEN,
+      UNINITIALIZED;
+
+      private State() {
+      }
+   }
+
    public static enum WorldType {
       NORMAL,
       MINIGAME,
       ADVENTUREMAP,
       EXPERIENCE,
@@ -305,19 +314,10 @@
 
       private WorldType() {
       }
    }
 
-   public static enum State {
-      CLOSED,
-      OPEN,
-      UNINITIALIZED;
-
-      private State() {
-      }
-   }
-
    public static class McoServerComparator implements Comparator<RealmsServer> {
       private final String refOwner;
 
       public McoServerComparator(String var1) {
          this.refOwner = var1;
