--- net/minecraft/world/level/levelgen/heightproviders/BiasedToBottomHeight.java
+++ net/minecraft/world/level/levelgen/heightproviders/BiasedToBottomHeight.java
@@ -48,8 +48,8 @@
    public HeightProviderType<?> getType() {
       return HeightProviderType.BIASED_TO_BOTTOM;
    }
 
    public String toString() {
-      return "biased[" + this.minInclusive + '-' + this.maxInclusive + " inner: " + this.inner + "]";
+      return "biased[" + this.minInclusive + "-" + this.maxInclusive + " inner: " + this.inner + "]";
    }
 }
