--- net/minecraft/client/gui/screens/ConfirmScreen.java
+++ net/minecraft/client/gui/screens/ConfirmScreen.java
@@ -1,10 +1,11 @@
 package net.minecraft.client.gui.screens;
 
 import com.mojang.blaze3d.vertex.PoseStack;
 import it.unimi.dsi.fastutil.booleans.BooleanConsumer;
 import java.util.Iterator;
+import java.util.Objects;
 import net.minecraft.client.gui.components.AbstractWidget;
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.components.MultiLineLabel;
 import net.minecraft.network.chat.CommonComponents;
 import net.minecraft.network.chat.Component;
@@ -31,18 +32,19 @@
       this.yesButton = var4;
       this.noButton = var5;
    }
 
    public String getNarrationMessage() {
-      return super.getNarrationMessage() + ". " + this.title2.getString();
+      String var10000 = super.getNarrationMessage();
+      return var10000 + ". " + this.title2.getString();
    }
 
    protected void init() {
       super.init();
       this.message = MultiLineLabel.create(this.font, this.title2, this.width - 50);
       int var10000 = this.message.getLineCount();
-      this.font.getClass();
+      Objects.requireNonNull(this.font);
       int var1 = var10000 * 9;
       int var2 = Mth.clamp(90 + var1 + 12, this.height / 6 + 96, this.height - 24);
       this.addButton(new Button(this.width / 2 - 155, var2, 150, 20, this.yesButton, (var1x) -> {
          this.callback.accept(true);
       }));
