--- net/minecraft/world/level/levelgen/feature/treedecorators/BeehiveDecorator.java
+++ net/minecraft/world/level/levelgen/feature/treedecorators/BeehiveDecorator.java
@@ -40,11 +40,11 @@
          }).collect(Collectors.toList());
          if (!var8.isEmpty()) {
             BlockPos var9 = (BlockPos)var8.get(var3.nextInt(var8.size()));
             BlockPos var10 = var9.relative(var6);
             if (Feature.isAir(var1, var10) && Feature.isAir(var1, var10.relative(Direction.SOUTH))) {
-               var2.accept(var10, Blocks.BEE_NEST.defaultBlockState().setValue(BeehiveBlock.FACING, Direction.SOUTH));
+               var2.accept(var10, (BlockState)Blocks.BEE_NEST.defaultBlockState().setValue(BeehiveBlock.FACING, Direction.SOUTH));
                var1.getBlockEntity(var10, BlockEntityType.BEEHIVE).ifPresent((var1x) -> {
                   int var2 = 2 + var3.nextInt(2);
 
                   for(int var3x = 0; var3x < var2; ++var3x) {
                      CompoundTag var4 = new CompoundTag();
