--- net/minecraft/nbt/ShortTag.java
+++ net/minecraft/nbt/ShortTag.java
@@ -24,11 +24,11 @@
          return true;
       }
    };
    private final short data;
 
-   private ShortTag(short var1) {
+   ShortTag(short var1) {
       this.data = var1;
    }
 
    public static ShortTag valueOf(short var0) {
       return var0 >= -128 && var0 <= 1024 ? ShortTag.Cache.cache[var0 - -128] : new ShortTag(var0);
