--- net/minecraft/world/level/storage/loot/functions/SetContainerLootTable.java
+++ net/minecraft/world/level/storage/loot/functions/SetContainerLootTable.java
@@ -11,14 +11,14 @@
 import net.minecraft.world.level.storage.loot.LootTable;
 import net.minecraft.world.level.storage.loot.ValidationContext;
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
 
 public class SetContainerLootTable extends LootItemConditionalFunction {
-   private final ResourceLocation name;
-   private final long seed;
+   final ResourceLocation name;
+   final long seed;
 
-   private SetContainerLootTable(LootItemCondition[] var1, ResourceLocation var2, long var3) {
+   SetContainerLootTable(LootItemCondition[] var1, ResourceLocation var2, long var3) {
       super(var1);
       this.name = var2;
       this.seed = var3;
    }
 
