--- net/minecraft/world/level/storage/loot/functions/SetEnchantmentsFunction.java
+++ net/minecraft/world/level/storage/loot/functions/SetEnchantmentsFunction.java
@@ -27,14 +27,14 @@
 import net.minecraft.world.level.storage.loot.parameters.LootContextParam;
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
 import net.minecraft.world.level.storage.loot.providers.number.NumberProvider;
 
 public class SetEnchantmentsFunction extends LootItemConditionalFunction {
-   private final Map<Enchantment, NumberProvider> enchantments;
-   private final boolean add;
+   final Map<Enchantment, NumberProvider> enchantments;
+   final boolean add;
 
-   private SetEnchantmentsFunction(LootItemCondition[] var1, Map<Enchantment, NumberProvider> var2, boolean var3) {
+   SetEnchantmentsFunction(LootItemCondition[] var1, Map<Enchantment, NumberProvider> var2, boolean var3) {
       super(var1);
       this.enchantments = ImmutableMap.copyOf(var2);
       this.add = var3;
    }
 
