--- net/minecraft/world/level/storage/loot/functions/SetItemDamageFunction.java
+++ net/minecraft/world/level/storage/loot/functions/SetItemDamageFunction.java
@@ -14,14 +14,14 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class SetItemDamageFunction extends LootItemConditionalFunction {
    private static final Logger LOGGER = LogManager.getLogger();
-   private final NumberProvider damage;
-   private final boolean add;
+   final NumberProvider damage;
+   final boolean add;
 
-   private SetItemDamageFunction(LootItemCondition[] var1, NumberProvider var2, boolean var3) {
+   SetItemDamageFunction(LootItemCondition[] var1, NumberProvider var2, boolean var3) {
       super(var1);
       this.damage = var2;
       this.add = var3;
    }
 
