--- net/minecraft/world/level/gameevent/EntityPositionSource.java
+++ net/minecraft/world/level/gameevent/EntityPositionSource.java
@@ -13,11 +13,11 @@
    public static final Codec<EntityPositionSource> CODEC = RecordCodecBuilder.create((var0) -> {
       return var0.group(Codec.INT.fieldOf("source_entity_id").forGetter((var0x) -> {
          return var0x.sourceEntityId;
       })).apply(var0, (Function)(EntityPositionSource::new));
    });
-   private final int sourceEntityId;
+   final int sourceEntityId;
    private Optional<Entity> sourceEntity = Optional.empty();
 
    public EntityPositionSource(int var1) {
       this.sourceEntityId = var1;
    }
