--- net/minecraft/server/packs/resources/ProfiledReloadInstance.java
+++ net/minecraft/server/packs/resources/ProfiledReloadInstance.java
@@ -68,17 +68,17 @@
 
       LOGGER.info((String)"Total blocking time: {} ms", (Object)var2);
    }
 
    public static class State {
-      private final String name;
-      private final ProfileResults preparationResult;
-      private final ProfileResults reloadResult;
-      private final AtomicLong preparationNanos;
-      private final AtomicLong reloadNanos;
+      final String name;
+      final ProfileResults preparationResult;
+      final ProfileResults reloadResult;
+      final AtomicLong preparationNanos;
+      final AtomicLong reloadNanos;
 
-      private State(String var1, ProfileResults var2, ProfileResults var3, AtomicLong var4, AtomicLong var5) {
+      State(String var1, ProfileResults var2, ProfileResults var3, AtomicLong var4, AtomicLong var5) {
          this.name = var1;
          this.preparationResult = var2;
          this.reloadResult = var3;
          this.preparationNanos = var4;
          this.reloadNanos = var5;
