--- net/minecraft/server/packs/resources/SimpleReloadableResourceManager.java
+++ net/minecraft/server/packs/resources/SimpleReloadableResourceManager.java
@@ -109,11 +109,11 @@
       this.listeners.add(var1);
    }
 
    public ReloadInstance createReload(Executor var1, Executor var2, CompletableFuture<Unit> var3, List<PackResources> var4) {
       LOGGER.info("Reloading ResourceManager: {}", () -> {
-         return (String)var4.stream().map(PackResources::getName).collect(Collectors.joining(", "));
+         return var4.stream().map(PackResources::getName).collect(Collectors.joining(", "));
       });
       this.clear();
       Iterator var5 = var4.iterator();
 
       while(var5.hasNext()) {
