--- net/minecraft/client/gui/screens/inventory/AbstractContainerScreen.java
+++ net/minecraft/client/gui/screens/inventory/AbstractContainerScreen.java
@@ -131,11 +131,11 @@
             var14.setCount(Mth.ceil((float)var14.getCount() / 2.0F));
          } else if (this.isQuickCrafting && this.quickCraftSlots.size() > 1) {
             var14 = var14.copy();
             var14.setCount(this.quickCraftingRemainder);
             if (var14.isEmpty()) {
-               var17 = "" + ChatFormatting.YELLOW + "0";
+               var17 = ChatFormatting.YELLOW + "0";
             }
          }
 
          this.renderFloatingItem(var14, var2 - var5 - 8, var3 - var6 - var10, var17);
       }
@@ -213,11 +213,12 @@
             var5 = var8.copy();
             var6 = true;
             AbstractContainerMenu.getQuickCraftSlotCount(this.quickCraftSlots, this.quickCraftingType, var5, var2.getItem().isEmpty() ? 0 : var2.getItem().getCount());
             int var10 = Math.min(var5.getMaxStackSize(), var2.getMaxStackSize(var5));
             if (var5.getCount() > var10) {
-               var9 = ChatFormatting.YELLOW.toString() + var10;
+               String var10000 = ChatFormatting.YELLOW.toString();
+               var9 = var10000 + var10;
                var5.setCount(var10);
             }
          } else {
             this.quickCraftSlots.remove(var2);
             this.recalculateQuickCraftRemaining();
@@ -227,11 +228,11 @@
       this.setBlitOffset(100);
       this.itemRenderer.blitOffset = 100.0F;
       if (var5.isEmpty() && var2.isActive()) {
          Pair var12 = var2.getNoItemIcon();
          if (var12 != null) {
-            TextureAtlasSprite var11 = (TextureAtlasSprite)this.minecraft.getTextureAtlas((ResourceLocation)var12.getFirst()).apply(var12.getSecond());
+            TextureAtlasSprite var11 = (TextureAtlasSprite)this.minecraft.getTextureAtlas((ResourceLocation)var12.getFirst()).apply((ResourceLocation)var12.getSecond());
             RenderSystem.setShaderTexture(0, var11.atlas().location());
             blit(var1, var3, var4, this.getBlitOffset(), 16, 16, var11);
             var7 = true;
          }
       }
