--- net/minecraft/client/gui/screens/inventory/AnvilScreen.java
+++ net/minecraft/client/gui/screens/inventory/AnvilScreen.java
@@ -41,11 +41,11 @@
       this.name = new EditBox(this.font, var1 + 62, var2 + 24, 103, 12, new TranslatableComponent("container.repair"));
       this.name.setCanLoseFocus(false);
       this.name.setTextColor(-1);
       this.name.setTextColorUneditable(-1);
       this.name.setBordered(false);
-      this.name.setMaxLength(35);
+      this.name.setMaxLength(50);
       this.name.setResponder(this::onNameChanged);
       this.name.setValue("");
       this.children.add(this.name);
       this.setInitialFocus(this.name);
       this.name.setEditable(false);
