--- net/minecraft/client/gui/screens/inventory/EffectRenderingInventoryScreen.java
+++ net/minecraft/client/gui/screens/inventory/EffectRenderingInventoryScreen.java
@@ -95,11 +95,11 @@
 
       for(Iterator var6 = var4.iterator(); var6.hasNext(); var5 += var3) {
          MobEffectInstance var7 = (MobEffectInstance)var6.next();
          String var8 = I18n.get(var7.getEffect().getDescriptionId());
          if (var7.getAmplifier() >= 1 && var7.getAmplifier() <= 9) {
-            var8 = var8 + ' ' + I18n.get("enchantment.level." + (var7.getAmplifier() + 1));
+            var8 = var8 + " " + I18n.get("enchantment.level." + (var7.getAmplifier() + 1));
          }
 
          this.font.drawShadow(var1, var8, (float)(var2 + 10 + 18), (float)(var5 + 6), 16777215);
          String var9 = MobEffectUtil.formatDuration(var7, 1.0F);
          this.font.drawShadow(var1, var9, (float)(var2 + 10 + 18), (float)(var5 + 6 + 10), 8355711);
