--- net/minecraft/client/gui/screens/inventory/MerchantScreen.java
+++ net/minecraft/client/gui/screens/inventory/MerchantScreen.java
@@ -44,11 +44,11 @@
    private static final Component TRADES_LABEL = new TranslatableComponent("merchant.trades");
    private static final Component LEVEL_SEPARATOR = new TextComponent(" - ");
    private static final Component DEPRECATED_TOOLTIP = new TranslatableComponent("merchant.deprecated");
    private int shopItem;
    private final MerchantScreen.TradeOfferButton[] tradeOfferButtons = new MerchantScreen.TradeOfferButton[7];
-   private int scrollOff;
+   int scrollOff;
    private boolean isDragging;
 
    public MerchantScreen(MerchantMenu var1, Inventory var2, Component var3) {
       super(var1, var2, var3);
       this.imageWidth = 276;
