--- net/minecraft/world/level/dimension/end/EndDragonFight.java
+++ net/minecraft/world/level/dimension/end/EndDragonFight.java
@@ -88,10 +88,14 @@
    public EndDragonFight(ServerLevel var1, long var2, CompoundTag var4) {
       this.dragonEvent = (ServerBossEvent)(new ServerBossEvent(new TranslatableComponent("entity.minecraft.ender_dragon"), BossEvent.BossBarColor.PINK, BossEvent.BossBarOverlay.PROGRESS)).setPlayBossMusic(true).setCreateWorldFog(true);
       this.gateways = Lists.newArrayList();
       this.needsStateScanning = true;
       this.level = var1;
+      if (var4.contains("NeedsStateScanning")) {
+         this.needsStateScanning = var4.getBoolean("NeedsStateScanning");
+      }
+
       if (var4.contains("DragonKilled", 99)) {
          if (var4.hasUUID("Dragon")) {
             this.dragonUUID = var4.getUUID("Dragon");
          }
 
@@ -123,10 +127,11 @@
       this.exitPortalPattern = BlockPatternBuilder.start().aisle("       ", "       ", "       ", "   #   ", "       ", "       ", "       ").aisle("       ", "       ", "       ", "   #   ", "       ", "       ", "       ").aisle("       ", "       ", "       ", "   #   ", "       ", "       ", "       ").aisle("  ###  ", " #   # ", "#     #", "#  #  #", "#     #", " #   # ", "  ###  ").aisle("       ", "  ###  ", " ##### ", " ##### ", " ##### ", "  ###  ", "       ").where('#', BlockInWorld.hasState(BlockPredicate.forBlock(Blocks.BEDROCK))).build();
    }
 
    public CompoundTag saveData() {
       CompoundTag var1 = new CompoundTag();
+      var1.putBoolean("NeedsStateScanning", this.needsStateScanning);
       if (this.dragonUUID != null) {
          var1.putUUID("Dragon", this.dragonUUID);
       }
 
       var1.putBoolean("DragonKilled", this.dragonKilled);
