--- net/minecraft/client/renderer/blockentity/SignRenderer.java
+++ net/minecraft/client/renderer/blockentity/SignRenderer.java
@@ -5,10 +5,11 @@
 import com.mojang.blaze3d.vertex.PoseStack;
 import com.mojang.blaze3d.vertex.VertexConsumer;
 import com.mojang.math.Vector3f;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.Font;
 import net.minecraft.client.model.Model;
 import net.minecraft.client.model.geom.EntityModelSet;
 import net.minecraft.client.model.geom.ModelLayers;
@@ -69,11 +70,11 @@
       }
 
       var3.pushPose();
       var3.scale(0.6666667F, -0.6666667F, -0.6666667F);
       Material var27 = Sheets.getSignMaterial(var9);
-      var10.getClass();
+      Objects.requireNonNull(var10);
       VertexConsumer var12 = var27.buffer(var4, var10::renderType);
       var10.root.render(var3, var12, var5, var6);
       var3.popPose();
       float var13 = 0.010416667F;
       var3.translate(0.0D, 0.3333333432674408D, 0.046666666865348816D);
