--- net/minecraft/world/level/storage/loot/providers/score/ContextScoreboardNameProvider.java
+++ net/minecraft/world/level/storage/loot/providers/score/ContextScoreboardNameProvider.java
@@ -12,13 +12,13 @@
 import net.minecraft.world.level.storage.loot.GsonAdapterFactory;
 import net.minecraft.world.level.storage.loot.LootContext;
 import net.minecraft.world.level.storage.loot.parameters.LootContextParam;
 
 public class ContextScoreboardNameProvider implements ScoreboardNameProvider {
-   private final LootContext.EntityTarget target;
+   final LootContext.EntityTarget target;
 
-   private ContextScoreboardNameProvider(LootContext.EntityTarget var1) {
+   ContextScoreboardNameProvider(LootContext.EntityTarget var1) {
       this.target = var1;
    }
 
    public static ScoreboardNameProvider forTarget(LootContext.EntityTarget var0) {
       return new ContextScoreboardNameProvider(var0);
