--- com/mojang/realmsclient/KeyCombo.java
+++ com/mojang/realmsclient/KeyCombo.java
@@ -38,8 +38,9 @@
    public void reset() {
       this.matchIndex = 0;
    }
 
    public String toString() {
-      return "KeyCombo{chars=" + Arrays.toString(this.chars) + ", matchIndex=" + this.matchIndex + '}';
+      String var10000 = Arrays.toString(this.chars);
+      return "KeyCombo{chars=" + var10000 + ", matchIndex=" + this.matchIndex + "}";
    }
 }
