--- net/minecraft/world/entity/ai/attributes/DefaultAttributes.java
+++ net/minecraft/world/entity/ai/attributes/DefaultAttributes.java
@@ -1,9 +1,10 @@
 package net.minecraft.world.entity.ai.attributes;
 
 import com.google.common.collect.ImmutableMap;
 import java.util.Map;
+import java.util.Objects;
 import java.util.stream.Stream;
 import net.minecraft.Util;
 import net.minecraft.core.DefaultedRegistry;
 import net.minecraft.core.Registry;
 import net.minecraft.world.entity.EntityType;
@@ -94,11 +95,11 @@
          return var0.getCategory() != MobCategory.MISC;
       }).filter((var0) -> {
          return !hasSupplier(var0);
       });
       DefaultedRegistry var10001 = Registry.ENTITY_TYPE;
-      var10001.getClass();
+      Objects.requireNonNull(var10001);
       var10000.map(var10001::getKey).forEach((var0) -> {
          Util.logAndPauseIfInIde("Entity " + var0 + " has no attributes");
       });
    }
 
