--- net/minecraft/server/commands/ExperienceCommand.java
+++ net/minecraft/server/commands/ExperienceCommand.java
@@ -96,11 +96,11 @@
 
          return var1.size();
       }
    }
 
-   static enum Type {
+   private static enum Type {
       POINTS("points", Player::giveExperiencePoints, (var0, var1) -> {
          if (var1 >= var0.getXpNeededForNextLevel()) {
             return false;
          } else {
             var0.setExperiencePoints(var1);
@@ -117,11 +117,11 @@
       });
 
       public final BiConsumer<ServerPlayer, Integer> add;
       public final BiPredicate<ServerPlayer, Integer> set;
       public final String name;
-      private final ToIntFunction<ServerPlayer> query;
+      final ToIntFunction<ServerPlayer> query;
 
       private Type(String var3, BiConsumer<ServerPlayer, Integer> var4, BiPredicate<ServerPlayer, Integer> var5, ToIntFunction<ServerPlayer> var6) {
          this.add = var4;
          this.name = var3;
          this.set = var5;
