--- net/minecraft/server/commands/GameRuleCommand.java
+++ net/minecraft/server/commands/GameRuleCommand.java
@@ -26,19 +26,19 @@
          }
       });
       var0.register(var1);
    }
 
-   private static <T extends GameRules.Value<T>> int setRule(CommandContext<CommandSourceStack> var0, GameRules.Key<T> var1) {
+   static <T extends GameRules.Value<T>> int setRule(CommandContext<CommandSourceStack> var0, GameRules.Key<T> var1) {
       CommandSourceStack var2 = (CommandSourceStack)var0.getSource();
       GameRules.Value var3 = var2.getServer().getGameRules().getRule(var1);
       var3.setFromArgument(var0, "value");
       var2.sendSuccess(new TranslatableComponent("commands.gamerule.set", new Object[]{var1.getId(), var3.toString()}), true);
       return var3.getCommandResult();
    }
 
-   private static <T extends GameRules.Value<T>> int queryRule(CommandSourceStack var0, GameRules.Key<T> var1) {
+   static <T extends GameRules.Value<T>> int queryRule(CommandSourceStack var0, GameRules.Key<T> var1) {
       GameRules.Value var2 = var0.getServer().getGameRules().getRule(var1);
       var0.sendSuccess(new TranslatableComponent("commands.gamerule.query", new Object[]{var1.getId(), var2.toString()}), false);
       return var2.getCommandResult();
    }
 }
