--- net/minecraft/server/commands/ScheduleCommand.java
+++ net/minecraft/server/commands/ScheduleCommand.java
@@ -72,11 +72,11 @@
             }
 
             var7.schedule(var8, var4, new FunctionTagCallback(var6));
             var0.sendSuccess(new TranslatableComponent("commands.schedule.created.tag", new Object[]{var6, var2, var4}), true);
          });
-         return (int)Math.floorMod(var4, 2147483647L);
+         return Math.floorMod(var4, Integer.MAX_VALUE);
       }
    }
 
    private static int remove(CommandSourceStack var0, String var1) throws CommandSyntaxException {
       int var2 = var0.getServer().getWorldData().overworldData().getScheduledEvents().remove(var1);
