--- com/mojang/realmsclient/client/RealmsClient.java
+++ com/mojang/realmsclient/client/RealmsClient.java
@@ -145,11 +145,11 @@
       String var2 = this.execute(Request.get(var1));
       return RealmsServerPlayerLists.parse(var2);
    }
 
    public RealmsServerAddress join(long var1) throws RealmsServiceException {
-      String var3 = this.url("worlds" + "/v1/$ID/join/pc".replace("$ID", "" + var1));
+      String var3 = this.url("worlds" + "/v1/$ID/join/pc".replace("$ID", var1.makeConcatWithConstants<invokedynamic>(var1)));
       String var4 = this.execute(Request.get(var3, 5000, 30000));
       return RealmsServerAddress.parse(var4);
    }
 
    public void initializeWorld(long var1, String var3, String var4) throws RealmsServiceException {
@@ -407,19 +407,10 @@
       currentEnvironment = RealmsClient.Environment.PRODUCTION;
       LOGGER = LogManager.getLogger();
       GSON = new GuardedSerializer();
    }
 
-   public static enum CompatibleVersionResponse {
-      COMPATIBLE,
-      OUTDATED,
-      OTHER;
-
-      private CompatibleVersionResponse() {
-      }
-   }
-
    public static enum Environment {
       PRODUCTION("pc.realms.minecraft.net", "https"),
       STAGE("pc-stage.realms.minecraft.net", "https"),
       LOCAL("localhost:8080", "http");
 
@@ -429,6 +420,15 @@
       private Environment(String var3, String var4) {
          this.baseUrl = var3;
          this.protocol = var4;
       }
    }
+
+   public static enum CompatibleVersionResponse {
+      COMPATIBLE,
+      OUTDATED,
+      OTHER;
+
+      private CompatibleVersionResponse() {
+      }
+   }
 }
