--- com/mojang/realmsclient/client/Request.java
+++ com/mojang/realmsclient/client/Request.java
@@ -194,10 +194,28 @@
       } catch (Exception var3) {
          return "";
       }
    }
 
+   public static class Get extends Request<Request.Get> {
+      public Get(String var1, int var2, int var3) {
+         super(var1, var2, var3);
+      }
+
+      public Request.Get doConnect() {
+         try {
+            this.connection.setDoInput(true);
+            this.connection.setDoOutput(true);
+            this.connection.setUseCaches(false);
+            this.connection.setRequestMethod("GET");
+            return this;
+         } catch (Exception var2) {
+            throw new RealmsHttpException(var2.getMessage(), var2);
+         }
+      }
+   }
+
    public static class Post extends Request<Request.Post> {
       private final String content;
 
       public Post(String var1, String var2, int var3, int var4) {
          super(var1, var3, var4);
@@ -224,10 +242,27 @@
             throw new RealmsHttpException(var3.getMessage(), var3);
          }
       }
    }
 
+   public static class Delete extends Request<Request.Delete> {
+      public Delete(String var1, int var2, int var3) {
+         super(var1, var2, var3);
+      }
+
+      public Request.Delete doConnect() {
+         try {
+            this.connection.setDoOutput(true);
+            this.connection.setRequestMethod("DELETE");
+            this.connection.connect();
+            return this;
+         } catch (Exception var2) {
+            throw new RealmsHttpException(var2.getMessage(), var2);
+         }
+      }
+   }
+
    public static class Put extends Request<Request.Put> {
       private final String content;
 
       public Put(String var1, String var2, int var3, int var4) {
          super(var1, var3, var4);
@@ -252,41 +287,6 @@
          } catch (Exception var3) {
             throw new RealmsHttpException(var3.getMessage(), var3);
          }
       }
    }
-
-   public static class Get extends Request<Request.Get> {
-      public Get(String var1, int var2, int var3) {
-         super(var1, var2, var3);
-      }
-
-      public Request.Get doConnect() {
-         try {
-            this.connection.setDoInput(true);
-            this.connection.setDoOutput(true);
-            this.connection.setUseCaches(false);
-            this.connection.setRequestMethod("GET");
-            return this;
-         } catch (Exception var2) {
-            throw new RealmsHttpException(var2.getMessage(), var2);
-         }
-      }
-   }
-
-   public static class Delete extends Request<Request.Delete> {
-      public Delete(String var1, int var2, int var3) {
-         super(var1, var2, var3);
-      }
-
-      public Request.Delete doConnect() {
-         try {
-            this.connection.setDoOutput(true);
-            this.connection.setRequestMethod("DELETE");
-            this.connection.connect();
-            return this;
-         } catch (Exception var2) {
-            throw new RealmsHttpException(var2.getMessage(), var2);
-         }
-      }
-   }
 }
