--- net/minecraft/world/level/levelgen/feature/configurations/OreConfiguration.java
+++ net/minecraft/world/level/levelgen/feature/configurations/OreConfiguration.java
@@ -56,11 +56,11 @@
          })).apply(var0, OreConfiguration.TargetBlockState::new);
       });
       public final RuleTest target;
       public final BlockState state;
 
-      private TargetBlockState(RuleTest var1, BlockState var2) {
+      TargetBlockState(RuleTest var1, BlockState var2) {
          this.target = var1;
          this.state = var2;
       }
    }
 
