--- net/minecraft/world/level/lighting/LayerLightEngine.java
+++ net/minecraft/world/level/lighting/LayerLightEngine.java
@@ -183,11 +183,12 @@
    public int getLightValue(BlockPos var1) {
       return this.storage.getLightValue(var1.asLong());
    }
 
    public String getDebugData(long var1) {
-      return "" + this.storage.getLevel(var1);
+      int var10000 = this.storage.getLevel(var1);
+      return var10000.makeConcatWithConstants<invokedynamic>(var10000);
    }
 
    public void checkBlock(BlockPos var1) {
       long var2 = var1.asLong();
       this.checkNode(var2);
