--- net/minecraft/world/level/lighting/SkyLightEngine.java
+++ net/minecraft/world/level/lighting/SkyLightEngine.java
@@ -177,11 +177,12 @@
       }
 
    }
 
    public String getDebugData(long var1) {
-      return super.getDebugData(var1) + (((SkyLightSectionStorage)this.storage).isAboveData(var1) ? "*" : "");
+      String var10000 = super.getDebugData(var1);
+      return var10000 + (((SkyLightSectionStorage)this.storage).isAboveData(var1) ? "*" : "");
    }
 
    static {
       HORIZONTALS = new Direction[]{Direction.NORTH, Direction.SOUTH, Direction.WEST, Direction.EAST};
    }
