--- net/minecraft/world/level/lighting/SpatialLongSet.java
+++ net/minecraft/world/level/lighting/SpatialLongSet.java
@@ -32,11 +32,11 @@
 
    public boolean isEmpty() {
       return this.map.isEmpty();
    }
 
-   public static class InternalMap extends Long2LongLinkedOpenHashMap {
+   protected static class InternalMap extends Long2LongLinkedOpenHashMap {
       private static final int X_BITS = Mth.log2(60000000);
       private static final int Z_BITS = Mth.log2(60000000);
       private static final int Y_BITS;
       private static final int Y_OFFSET = 0;
       private static final int Z_OFFSET;
