--- net/minecraft/world/level/storage/loot/functions/SetItemCountFunction.java
+++ net/minecraft/world/level/storage/loot/functions/SetItemCountFunction.java
@@ -11,14 +11,14 @@
 import net.minecraft.world.level.storage.loot.parameters.LootContextParam;
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
 import net.minecraft.world.level.storage.loot.providers.number.NumberProvider;
 
 public class SetItemCountFunction extends LootItemConditionalFunction {
-   private final NumberProvider value;
-   private final boolean add;
+   final NumberProvider value;
+   final boolean add;
 
-   private SetItemCountFunction(LootItemCondition[] var1, NumberProvider var2, boolean var3) {
+   SetItemCountFunction(LootItemCondition[] var1, NumberProvider var2, boolean var3) {
       super(var1);
       this.value = var2;
       this.add = var3;
    }
 
